// @progress/kendo-vue-scheduler/dist/npm/components/footer/bussiness-hours/BusinessHours.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BusinessHoursVue2 = exports.BusinessHours = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var main_1 = require("../../../messages/main");
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var kendo_vue_buttons_2 = require("@progress/kendo-vue-buttons");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var kendo_svg_icons_1 = require("@progress/kendo-svg-icons");
/**
 * @hidden
 */
var BusinessHoursVue2 = {
  name: 'KendoBusinessHours',
  // @ts-ignore
  emits: {
    click: null
  },
  inject: {
    ks: {
      default: null
    },
    kendoLocalizationService: {
      default: null
    }
  },
  props: {
    slotDuration: Number
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var _this = this;
    var h = gh || createElement;
    var defaultSlot = (0, kendo_vue_common_1.getDefaultSlots)(this);
    this.localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    return this.slotDuration && this.slotDuration < 24 * 60 ?
    // @ts-ignore  function children
    h(kendo_vue_buttons_1.ToolbarItem, {
      // ref={(item) => { if (item) { element.current = item.element; } }}
      "class": 'k-scheduler-navigation'
    }, this.v3 ? function () {
      return [defaultSlot ? defaultSlot :
      // @ts-ignore  function children
      h(kendo_vue_buttons_2.Button, {
        tabIndex: -1,
        attrs: _this.v3 ? undefined : {
          tabIndex: -1,
          icon: 'clock',
          svgIcon: kendo_svg_icons_1.clockIcon
        },
        icon: 'clock',
        svgIcon: kendo_svg_icons_1.clockIcon,
        onClick: _this.handleShowWorkHoursClick,
        on: _this.v3 ? undefined : {
          "click": _this.handleShowWorkHoursClick
        }
      }, _this.v3 ? function () {
        return [_this.localization.toLanguageString(_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay, main_1.messages[_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay])];
      } : [_this.localization.toLanguageString(_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay, main_1.messages[_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay])])];
    } : [defaultSlot ? defaultSlot : h(kendo_vue_buttons_2.Button, {
      tabIndex: -1,
      attrs: _this.v3 ? undefined : {
        tabIndex: -1,
        icon: 'clock',
        svgIcon: kendo_svg_icons_1.clockIcon
      },
      icon: 'clock',
      svgIcon: kendo_svg_icons_1.clockIcon,
      onClick: _this.handleShowWorkHoursClick,
      on: _this.v3 ? undefined : {
        "click": _this.handleShowWorkHoursClick
      }
    }, _this.v3 ? function () {
      return [_this.localization.toLanguageString(_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay, main_1.messages[_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay])];
    } : [_this.localization.toLanguageString(_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay, main_1.messages[_this.ks.showWorkHours ? main_1.showFullDay : main_1.showWorkDay])])]) : null;
  },
  methods: {
    handleShowWorkHoursClick: function handleShowWorkHoursClick(e) {
      this.$emit('click', e);
    }
  }
};
exports.BusinessHoursVue2 = BusinessHoursVue2;
/**
 * @hidden
 */
var BusinessHours = BusinessHoursVue2;
exports.BusinessHours = BusinessHours;
