// @ckeditor/vite-plugin-ckeditor5/dist/index.cjs
'use strict';

const fs = require('node:fs');
const path = require('node:path');
const util = require('node:util');
const vite = require('vite');
const postcss = require('postcss');
const postcssNesting = require('postcss-nesting');
const postcssMixins = require('postcss-mixins');
const postcssImport = require('postcss-import');

function _interopDefaultCompat (e) { return e && typeof e === 'object' && 'default' in e ? e.default : e; }

function _interopNamespaceCompat(e) {
	if (e && typeof e === 'object' && 'default' in e) return e;
	const n = Object.create(null);
	if (e) {
		for (const k in e) {
			n[k] = e[k];
		}
	}
	n.default = e;
	return n;
}

const fs__namespace = /*#__PURE__*/_interopNamespaceCompat(fs);
const path__namespace = /*#__PURE__*/_interopNamespaceCompat(path);
const util__namespace = /*#__PURE__*/_interopNamespaceCompat(util);
const postcss__default = /*#__PURE__*/_interopDefaultCompat(postcss);
const postcssNesting__default = /*#__PURE__*/_interopDefaultCompat(postcssNesting);
const postcssMixins__default = /*#__PURE__*/_interopDefaultCompat(postcssMixins);
const postcssImport__default = /*#__PURE__*/_interopDefaultCompat(postcssImport);

/**
 * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
 * For licensing, see LICENSE.md.
 */
const isCKEditor5Icon = (id) => id.endsWith(".svg") && (id.includes("ckeditor5-") || id.includes("-ckeditor5"));
const iconsLoader = () => {
  return {
    name: "vite-ckeditor5-icons-loader",
    enforce: "pre",
    load: (id) => {
      if (!isCKEditor5Icon(id)) {
        return;
      }
      return fs__namespace.readFileSync(id, "utf8");
    },
    transform: (code, id) => {
      if (!isCKEditor5Icon(id)) {
        return;
      }
      const content = JSON.stringify(code);
      return {
        code: `export default ${content};`,
        map: { mappings: "" }
      };
    }
  };
};

/**
 * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
 * For licensing, see LICENSE.md.
 */
const readFile = util__namespace.promisify(fs__namespace.readFile);
const postcssOptions = {
  plugins: [
    postcssNesting__default(),
    postcssMixins__default(),
    postcssImport__default()
  ]
};
const stylesLoader = (theme) => {
  return {
    name: "vite-ckeditor5-styles-loader",
    enforce: "pre",
    async transform(code, id) {
      if (!id.includes("ckeditor5-") || !id.endsWith(".css")) {
        return;
      }
      const themeStyles = await loadThemeStyles(id, theme);
      const stylesToLoad = code + themeStyles.code;
      return loadPostcssFile(stylesToLoad, id);
    }
  };
};
async function loadThemeStyles(id, theme) {
  const themeFilePath = getThemeFilePath(id, theme);
  if (themeFilePath && fs__namespace.existsSync(themeFilePath)) {
    const css = await readFile(themeFilePath);
    const result = await postcss__default(postcssOptions).process(css, { from: themeFilePath, map: { absolute: true, inline: true } });
    return { code: result.css.toString() };
  }
  return { code: "" };
}
async function loadPostcssFile(code, id) {
  const result = await postcss__default(postcssOptions).process(code, { from: id, map: { absolute: true, inline: true } });
  return {
    code: result.css.toString()
  };
}
function getThemeFilePath(inputFilePath, theme) {
  const themeDir = path__namespace.dirname(theme);
  const packageName = getPackageName(inputFilePath);
  if (!packageName) {
    return;
  }
  const inputFileName = inputFilePath.split(
    vite.normalizePath(path__namespace.join(packageName, "theme", path__namespace.sep))
  )[1];
  if (!inputFileName) {
    return;
  }
  return path__namespace.resolve(themeDir, packageName, inputFileName);
}
function getPackageName(inputFilePath) {
  const match = inputFilePath.match(/^.+[/\\](ckeditor5-[^/\\]+)/);
  if (match) {
    return match.pop();
  } else {
    return null;
  }
}

/**
 * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
 * For licensing, see LICENSE.md.
 */
const ckeditor5 = ({ theme }) => {
  return [
    iconsLoader(),
    stylesLoader(theme)
  ];
};

module.exports = ckeditor5;
