// @progress/kendo-vue-scheduler/dist/es/views/time/TimelineViewAllEventsRowContent.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
import { messages, allEvents } from '../../messages/main';
import { provideLocalizationService } from '@progress/kendo-vue-intl';
/** @hidden */
var TimelineViewAllEventsRowContentVue2 = {
  name: 'KendoTimelineViewAllEventsRowContent',
  props: {},
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var localization = provideLocalizationService(this);
    return h("div", {
      "class": "k-sticky-cell"
    }, [h("div", {
      "class": "k-scheduler-cell k-heading-cell k-side-cell k-scheduler-times-all-day",
      "data-range-index": 0,
      attrs: this.v3 ? undefined : {
        "data-range-index": 0
      }
    }, [localization.toLanguageString(allEvents, messages[allEvents])])]);
  }
};
/** @hidden */
var TimelineViewAllEventsRowContent = TimelineViewAllEventsRowContentVue2;
export { TimelineViewAllEventsRowContent, TimelineViewAllEventsRowContentVue2 };
