// @progress/kendo-vue-scheduler/dist/es/views/time/TimelineView.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
import { MultiDayTimelineView } from './MultiDayTimelineView';
import { Day } from '@progress/kendo-date-math';
/** @hidden */
var TimelineViewVue2 = {
  name: 'KendoTimelineView',
  // @ts-ignore
  emits: {
    dataaction: null
  },
  props: {
    name: {
      type: String,
      default: 'timeline'
    },
    title: {
      type: String,
      default: 'Multi Day Timeline'
    },
    currentTimeMarker: {
      type: Boolean,
      default: true
    },
    dateRange: {
      type: [Function, Object],
      default: function _default() {
        return MultiDayTimelineView.props.dateRange.default();
      }
    },
    showWorkHours: {
      type: Boolean,
      default: true
    },
    selectedDateFormat: {
      type: String,
      default: '{0:D}'
    },
    selectedShortDateFormat: {
      type: String,
      default: '{0:d}'
    },
    step: {
      type: Number,
      default: 1
    },
    numberOfDays: {
      type: Number,
      default: 1
    },
    startTime: {
      type: String,
      default: '00:00'
    },
    endTime: {
      type: String,
      default: '00:00'
    },
    workDayStart: {
      type: String,
      default: '8:00'
    },
    workDayEnd: {
      type: String,
      default: '17:00'
    },
    workWeekStart: {
      type: Number,
      default: Day.Monday
    },
    workWeekEnd: {
      type: Number,
      default: Day.Friday
    },
    slotDivisions: {
      type: Number,
      default: 2
    },
    slotDuration: {
      type: Number,
      default: 60
    },
    defaultShowWorkHours: {
      type: Boolean,
      default: true
    },
    columnWidth: {
      type: Number,
      default: 100
    },
    dragItem: {
      type: Object,
      default: function _default() {
        return undefined;
      }
    },
    resizeItem: {
      type: Object,
      default: function _default() {
        return undefined;
      }
    },
    dateHeaderCell: [String, Function, Object],
    timeHeaderCell: [String, Function, Object],
    slotRender: [String, Function, Object],
    viewSlot: [String, Function, Object],
    resizeHint: [String, Function, Object],
    item: [String, Function, Object],
    viewItem: [String, Function, Object],
    editable: [Object, Boolean]
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    return h(MultiDayTimelineView, {
      name: this.$props.name,
      attrs: this.v3 ? undefined : {
        name: this.$props.name,
        dragItem: this.$props.dragItem,
        resizeItem: this.$props.resizeItem,
        dateRange: this.$props.dateRange,
        step: this.$props.step,
        numberOfDays: this.$props.numberOfDays,
        slotDuration: this.$props.slotDuration,
        slotDivisions: this.$props.slotDivisions,
        selectedDateFormat: this.$props.selectedDateFormat,
        selectedShortDateFormat: this.$props.selectedShortDateFormat,
        showWorkHours: this.$props.showWorkHours,
        columnWidth: this.$props.columnWidth,
        title: this.$props.title,
        currentTimeMarker: this.$props.currentTimeMarker,
        startTime: this.$props.startTime,
        endTime: this.$props.endTime,
        workDayStart: this.$props.workDayStart,
        workDayEnd: this.$props.workDayEnd,
        workWeekStart: this.$props.workWeekStart,
        workWeekEnd: this.$props.workWeekEnd,
        timeHeaderCell: this.$props.timeHeaderCell,
        dateHeaderCell: this.$props.dateHeaderCell,
        slotRender: this.$props.slotRender,
        viewSlot: this.$props.viewSlot,
        resizeHint: this.$props.resizeHint,
        item: this.$props.item,
        viewItem: this.$props.viewItem,
        editable: this.$props.editable
      },
      dragItem: this.$props.dragItem,
      resizeItem: this.$props.resizeItem,
      dateRange: this.$props.dateRange,
      step: this.$props.step,
      numberOfDays: this.$props.numberOfDays,
      slotDuration: this.$props.slotDuration,
      slotDivisions: this.$props.slotDivisions,
      selectedDateFormat: this.$props.selectedDateFormat,
      selectedShortDateFormat: this.$props.selectedShortDateFormat,
      showWorkHours: this.$props.showWorkHours,
      columnWidth: this.$props.columnWidth,
      title: this.$props.title,
      currentTimeMarker: this.$props.currentTimeMarker,
      startTime: this.$props.startTime,
      endTime: this.$props.endTime,
      workDayStart: this.$props.workDayStart,
      workDayEnd: this.$props.workDayEnd,
      workWeekStart: this.$props.workWeekStart,
      workWeekEnd: this.$props.workWeekEnd,
      timeHeaderCell: this.$props.timeHeaderCell,
      dateHeaderCell: this.$props.dateHeaderCell,
      slotRender: this.$props.slotRender,
      viewSlot: this.$props.viewSlot,
      resizeHint: this.$props.resizeHint,
      item: this.$props.item,
      viewItem: this.$props.viewItem,
      editable: this.$props.editable,
      onDataaction: this.handleDataAction,
      on: this.v3 ? undefined : {
        "dataaction": this.handleDataAction
      }
    });
  },
  methods: {
    handleDataAction: function handleDataAction(action) {
      this.$emit('dataaction', action);
    }
  }
};
/** @hidden */
var TimelineView = TimelineViewVue2;
export { TimelineView, TimelineViewVue2 };
