// @progress/kendo-vue-notification/dist/npm/Notification.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NotificationVue2 = exports.Notification = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var package_metadata_1 = require("./package-metadata");
var kendo_svg_icons_1 = require("@progress/kendo-svg-icons");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var buttonTitle = 'Close';
/**
 * @hidden
 */
var NotificationVue2 = {
  name: 'KendoNotification',
  props: {
    closable: {
      type: [Object, Boolean],
      default: function _default() {
        return false;
      }
    },
    dir: String,
    type: {
      type: Object,
      default: function _default() {
        return {
          style: 'none',
          icon: true
        };
      },
      validator: function validator(value) {
        return ['none', 'success', 'error', 'warning', 'info'].includes(value.style);
      }
    },
    themeColor: {
      type: String,
      default: 'base',
      validator: function validator(value) {
        return [null, 'base', 'error', 'info', 'success', 'warning'].includes(value);
      }
    },
    ariaLabel: {
      type: String,
      default: undefined
    }
  },
  created: function created() {
    (0, kendo_vue_common_1.validatePackage)(package_metadata_1.packageMetadata);
    this.notificationContentId = (0, kendo_vue_common_1.guid)();
  },
  computed: {
    computedTheme: function computedTheme() {
      var _a = this.$props,
        type = _a.type,
        themeColor = _a.themeColor;
      return type.style !== 'none' ? type.style : themeColor;
    },
    computedSvg: function computedSvg() {
      switch (this.computedTheme) {
        case 'success':
          return {
            name: 'check-outline',
            icon: kendo_svg_icons_1.checkOutlineIcon
          };
        case 'error':
          return {
            name: 'x-outline',
            icon: kendo_svg_icons_1.xOutlineIcon
          };
        case 'info':
          return {
            name: 'info-circle',
            icon: kendo_svg_icons_1.infoCircleIcon
          };
        case 'warning':
          return {
            name: 'exclamation-circle',
            icon: kendo_svg_icons_1.exclamationCircleIcon
          };
        default:
          return {};
      }
    },
    wrapperClass: function wrapperClass() {
      var _a;
      var closable = this.$props.closable;
      return _a = {
        'k-notification': true
      }, _a["k-notification-".concat(this.computedTheme)] = this.computedTheme, _a['k-notification-closable'] = closable, _a;
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlot = (0, kendo_vue_common_1.getDefaultSlots)(this);
    var _a = this.$props,
      closable = _a.closable,
      type = _a.type;
    return h("div", {
      "class": "k-notification-container"
    }, [h("div", {
      role: 'alert',
      attrs: this.v3 ? undefined : {
        role: 'alert',
        "aria-live": 'polite',
        "aria-label": this.$props.ariaLabel,
        "aria-describedby": this.notificationContentId
      },
      "class": this.wrapperClass,
      style: this.$props.style,
      "aria-live": 'polite',
      "aria-label": this.$props.ariaLabel,
      "aria-describedby": this.notificationContentId
    }, [type.icon && this.computedSvg.name && h(kendo_vue_common_1.Icon, {
      name: this.computedSvg.name,
      attrs: this.v3 ? undefined : {
        name: this.computedSvg.name,
        icon: this.computedSvg.icon
      },
      icon: this.computedSvg.icon,
      "class": 'k-notification-status'
    }), h("div", {
      "class": "k-notification-content",
      id: this.notificationContentId,
      attrs: this.v3 ? undefined : {
        id: this.notificationContentId
      }
    }, [defaultSlot]), closable && h("span", {
      "class": "k-notification-actions"
    }, [h("span", {
      "class": "k-notification-action k-notification-close-action",
      "aria-hidden": true,
      attrs: this.v3 ? undefined : {
        "aria-hidden": true
      }
    }, [h(kendo_vue_common_1.Icon, {
      name: "x",
      attrs: this.v3 ? undefined : {
        name: "x",
        icon: kendo_svg_icons_1.xIcon,
        title: buttonTitle
      },
      icon: kendo_svg_icons_1.xIcon,
      title: buttonTitle,
      onClick: this.handleClick,
      on: this.v3 ? undefined : {
        "click": this.handleClick
      }
    })])])])]);
  },
  methods: {
    handleClick: function handleClick(event) {
      this.$emit('close', {
        component: this,
        event: event
      });
    }
  }
};
exports.NotificationVue2 = NotificationVue2;
/**
 * Represents the [Kendo UI for Vue Native Notification component]({% slug overview_notification %}).
 *
 *
 * ### props <span class='code'>Readonly&lt;[NotificationProps]({% slug api_notification_notificationprops %})</span>
 * The props of the Notification component.
 *
 */
var Notification = NotificationVue2;
exports.Notification = Notification;
