// @progress/kendo-vue-editor/dist/npm/tools/inlineFormat.js
"use strict";

var __assign = undefined && undefined.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];
      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
var __rest = undefined && undefined.__rest || function (s, e) {
  var t = {};
  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }
  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InlineFormatVue2 = exports.InlineFormat = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_editor_common_1 = require("@progress/kendo-editor-common");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var main_1 = require("../messages/main");
/**
 * @hidden
 */
var InlineFormatVue2 = {
  name: 'KendoInlineFormat',
  props: __assign({
    view: Object,
    settings: {
      type: Object,
      default: function _default() {
        return {
          messages: {}
        };
      }
    }
  }, kendo_vue_buttons_1.Button.props),
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
      view = _a.view,
      render = _a.render,
      settings = _a.settings,
      other = __rest(_a, ["view", "render", "settings"]);
    var state = view && view.state;
    var mark = settings.mark,
      altMarks = settings.altMarks,
      altStyle = settings.altStyle;
    var options = {
      mark: mark,
      altMarks: altMarks,
      altStyle: altStyle
    };
    var localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    var titleKey = settings.messages.title;
    var buttonProps = __assign(__assign(__assign({}, other), {
      selected: !!state && (0, kendo_editor_common_1.hasMark)(state, options),
      togglable: true,
      title: localization.toLanguageString(titleKey, main_1.messages[titleKey])
    }), settings.props);
    var button = h(kendo_vue_buttons_1.Button, __assign({
      onClick: this.handleClick,
      onMousedown: this.preventDefault,
      onPointerdown: this.preventDefault,
      on: this.v3 ? undefined : {
        'click': this.handleClick,
        'mousedown': this.preventDefault,
        'pointerdown': this.preventDefault
      },
      attrs: this.v3 ? undefined : buttonProps
    }, buttonProps));
    return button;
  },
  methods: {
    handleClick: function handleClick() {
      var state = this.view && this.view.state;
      var _a = this.settings,
        mark = _a.mark,
        altMarks = _a.altMarks,
        altStyle = _a.altStyle;
      var options = {
        mark: mark,
        altMarks: altMarks,
        altStyle: altStyle
      };
      return this.view && (0, kendo_editor_common_1.toggleInlineFormat)(options, state && state.tr.setMeta('commandName', this.settings.commandName))(this.view.state, this.view.dispatch);
    },
    preventDefault: function preventDefault(e) {
      e.preventDefault();
    }
  }
};
exports.InlineFormatVue2 = InlineFormatVue2;
/**
 * @hidden
 */
var InlineFormat = InlineFormatVue2;
exports.InlineFormat = InlineFormat;
