// @progress/kendo-vue-editor/dist/npm/tools/pdf.js
"use strict";

var __assign = undefined && undefined.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];
      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
var __rest = undefined && undefined.__rest || function (s, e) {
  var t = {};
  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }
  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PdfVue2 = exports.Pdf = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var kendo_vue_pdf_1 = require("@progress/kendo-vue-pdf");
var main_1 = require("../messages/main");
var toolsSettings_1 = require("./../config/toolsSettings");
var pdf = toolsSettings_1.EditorToolsSettings.pdf,
  defaultSavePdfOptions = toolsSettings_1.EditorToolsSettings.savePdfOptions;
/**
 * @hidden
 */
var PdfVue2 = {
  name: 'KendoPdf',
  props: __assign({
    view: Object,
    settings: Object
  }, kendo_vue_buttons_1.Button.props),
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
      view = _a.view,
      render = _a.render,
      _b = _a.settings,
      settings = _b === void 0 ? pdf : _b,
      savePdfOptions = _a.savePdfOptions,
      savePdfCallback = _a.savePdfCallback,
      other = __rest(_a, ["view", "render", "settings", "savePdfOptions", "savePdfCallback"]);
    var titleKey = settings.messages.title;
    var localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    var buttonProps = __assign(__assign(__assign({}, other), {
      title: localization.toLanguageString(titleKey, main_1.messages[titleKey])
    }), settings.props);
    var button = h(kendo_vue_buttons_1.Button, __assign({
      onClick: this.handleClick,
      onMousedown: this.preventDefault,
      onPointerdown: this.preventDefault,
      on: this.v3 ? undefined : {
        'click': this.handleClick,
        'mousedown': this.preventDefault,
        'pointerdown': this.preventDefault
      },
      attrs: this.v3 ? undefined : buttonProps
    }, buttonProps));
    return button;
  },
  methods: {
    handleClick: function handleClick() {
      if (this.view) {
        (0, kendo_vue_pdf_1.savePDF)(this.view.dom, __assign(__assign({}, defaultSavePdfOptions), this.savePdfOptions || {}), this.savePdfCallback);
      }
    },
    preventDefault: function preventDefault(e) {
      e.preventDefault();
    }
  }
};
exports.PdfVue2 = PdfVue2;
/**
 * @hidden
 */
var Pdf = PdfVue2;
exports.Pdf = Pdf;
