// @progress/kendo-vue-scheduler/dist/npm/components/BaseView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BaseViewVue2 = exports.BaseView = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var main_1 = require("../hooks/main");
var use_items_selection_1 = require("../hooks/use-items-selection");
var use_slots_selection_1 = require("../hooks/use-slots-selection");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var main_2 = require("../utils/main");
/**
 * @hidden
 */
var BaseViewVue2 = {
  name: 'KendoBaseView',
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  props: {
    viewProps: Object,
    baseSlots: Array,
    ranges: Array,
    id: String,
    isAgendaView: {
      type: Boolean,
      default: false
    }
  },
  provide: function provide() {
    return {
      bv: this.bv,
      setItems: this.dispatchItems,
      setSlots: this.dispatchSlots,
      dispatchSelectedSlots: this.handleSchedulerSlotsSelectAction,
      dispatchSelectedItems: this.handleSchedulerItemsSelectAction,
      dispatchFocusedItems: this.setFocusedItems,
      dispatchFocusedSlots: this.setFocusedSlots
      //    setActiveViewName: this.handleActiveViewNameChange
    };
  },

  data: function data() {
    return {
      bv: {
        viewProps: this.viewProps,
        ranges: this.ranges,
        items: [],
        slots: [],
        focusedItems: [],
        focusedSlots: [],
        selectedItems: [],
        selectedSlots: []
      }
    };
  },
  mounted: function mounted() {
    this.syncCells();
  },
  updated: function updated() {
    this.syncCells();
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlot = (0, kendo_vue_common_1.getDefaultSlots)(this);
    return h("div", {
      "class": 'k-scheduler-layout k-scheduler-layout-flex',
      role: this.$props.isAgendaView ? 'grid' : 'presentation',
      attrs: this.v3 ? undefined : {
        role: this.$props.isAgendaView ? 'grid' : 'presentation'
      }
    }, [defaultSlot]);
  },
  methods: {
    syncCells: function syncCells() {
      (0, main_1.useCellSync)({
        element: this.$el,
        selector: '.k-group-cell',
        attribute: 'data-depth-index',
        explicitDepth: true
      });
      (0, main_1.useCellSync)({
        element: this.$el,
        selector: '.k-side-cell',
        attribute: 'data-depth-index',
        explicitDepth: false
      });
    },
    handleSchedulerItemsSelectAction: function handleSchedulerItemsSelectAction(action, event) {
      use_slots_selection_1.dispatchSelectedSlots.call(this, {
        type: use_slots_selection_1.SLOTS_SELECT_ACTION.reset
      });
      use_items_selection_1.dispatchSelectedItems.call(this, action, event);
    },
    handleSchedulerSlotsSelectAction: function handleSchedulerSlotsSelectAction(action, event) {
      use_items_selection_1.dispatchSelectedItems.call(this, {
        type: use_items_selection_1.ITEMS_SELECT_ACTION.reset
      }, event);
      use_slots_selection_1.dispatchSelectedSlots.call(this, action);
    },
    setFocusedItems: function setFocusedItems(action, event) {
      main_1.dispatchFocusedItems.call(this, action, event);
    },
    setFocusedSlots: function setFocusedSlots(action, event) {
      main_1.dispatchFocusedSlots.call(this, action, event);
    },
    dispatchItems: function dispatchItems(action) {
      switch (action.type) {
        case main_2.COLLECTION_ACTION.add:
          this.bv.items.push(action.item);
          break;
        case main_2.COLLECTION_ACTION.remove:
          var index = this.bv.items.indexOf(action.item);
          this.bv.items.splice(index, 1);
          break;
        default:
          break;
      }
    },
    dispatchSlots: function dispatchSlots(action) {
      switch (action.type) {
        case main_2.COLLECTION_ACTION.add:
          this.bv.slots.push(action.item);
          break;
        case main_2.COLLECTION_ACTION.remove:
          var index = this.bv.slots.indexOf(action.item);
          this.bv.slots.splice(index, 1);
          break;
        default:
          break;
      }
    }
  }
};
exports.BaseViewVue2 = BaseViewVue2;
/**
 * @hidden
 */
var BaseView = BaseViewVue2;
exports.BaseView = BaseView;
