// @progress/kendo-vue-scheduler/dist/npm/components/SchedulerRemoveDialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SchedulerRemoveDialogVue2 = exports.SchedulerRemoveDialog = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var kendo_vue_dialogs_1 = require("@progress/kendo-vue-dialogs");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var main_1 = require("../messages/main");
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
/**
 * @hidden
 */
var SchedulerRemoveDialogVue2 = {
  name: 'KendoSchedulerRemoveDialog',
  props: {
    dataItem: Object
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  created: function created() {
    this.dialogId = (0, kendo_vue_common_1.guid)();
  },
  mounted: function mounted() {
    var that = this;
    setTimeout(function () {
      var dialogWrapper = document.getElementById(that.dialogId);
      if (dialogWrapper) {
        dialogWrapper.focus();
      }
    }, 10);
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    var kendoLocalizationService = inject('kendoLocalizationService', {});
    return {
      v3: v3,
      kendoLocalizationService: kendoLocalizationService
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var _this = this;
    var h = gh || createElement;
    var localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    var removeCancelMessage = localization.toLanguageString(main_1.editorCancel, main_1.messages[main_1.editorCancel]);
    var removeConfirmMessage = localization.toLanguageString(main_1.editorDelete, main_1.messages[main_1.editorDelete]);
    var removeMessage = localization.toLanguageString(main_1.deleteConfirmation, main_1.messages[main_1.deleteConfirmation]);
    var deleteDialogTitleMessage = localization.toLanguageString(main_1.deleteDialogTitle, main_1.messages[main_1.deleteDialogTitle]);
    return (
      // @ts-ignore  function children
      h(kendo_vue_dialogs_1.Dialog, {
        title: deleteDialogTitleMessage,
        attrs: this.v3 ? undefined : {
          title: deleteDialogTitleMessage,
          appendTo: 'body',
          wrapperId: this.dialogId
        },
        appendTo: 'body',
        wrapperId: this.dialogId,
        onClose: this.handleClose,
        on: this.v3 ? undefined : {
          "close": this.handleClose
        }
      }, this.v3 ? function () {
        return [removeMessage,
        // @ts-ignore  function children
        h(kendo_vue_dialogs_1.DialogActionsBar, _this.v3 ? function () {
          return [
          // @ts-ignore  function children
          h(kendo_vue_buttons_1.Button, {
            onClick: _this.handleCancel,
            on: _this.v3 ? undefined : {
              "click": _this.handleCancel
            }
          }, _this.v3 ? function () {
            return [removeCancelMessage];
          } : [removeCancelMessage]),
          // @ts-ignore  function children
          h(kendo_vue_buttons_1.Button, {
            onClick: _this.handleConfirm,
            on: _this.v3 ? undefined : {
              "click": _this.handleConfirm
            }
          }, _this.v3 ? function () {
            return [removeConfirmMessage];
          } : [removeConfirmMessage])];
        } : [h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleCancel,
          on: _this.v3 ? undefined : {
            "click": _this.handleCancel
          }
        }, _this.v3 ? function () {
          return [removeCancelMessage];
        } : [removeCancelMessage]), h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleConfirm,
          on: _this.v3 ? undefined : {
            "click": _this.handleConfirm
          }
        }, _this.v3 ? function () {
          return [removeConfirmMessage];
        } : [removeConfirmMessage])])];
      } : [removeMessage, h(kendo_vue_dialogs_1.DialogActionsBar, _this.v3 ? function () {
        return [h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleCancel,
          on: _this.v3 ? undefined : {
            "click": _this.handleCancel
          }
        }, _this.v3 ? function () {
          return [removeCancelMessage];
        } : [removeCancelMessage]), h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleConfirm,
          on: _this.v3 ? undefined : {
            "click": _this.handleConfirm
          }
        }, _this.v3 ? function () {
          return [removeConfirmMessage];
        } : [removeConfirmMessage])];
      } : [h(kendo_vue_buttons_1.Button, {
        onClick: _this.handleCancel,
        on: _this.v3 ? undefined : {
          "click": _this.handleCancel
        }
      }, _this.v3 ? function () {
        return [removeCancelMessage];
      } : [removeCancelMessage]), h(kendo_vue_buttons_1.Button, {
        onClick: _this.handleConfirm,
        on: _this.v3 ? undefined : {
          "click": _this.handleConfirm
        }
      }, _this.v3 ? function () {
        return [removeConfirmMessage];
      } : [removeConfirmMessage])])])
    );
  },
  methods: {
    handleClose: function handleClose(_a) {
      var event = _a.event;
      this.$emit('close', {
        event: event
      });
    },
    handleCancel: function handleCancel(event) {
      this.$emit('cancel', {
        event: event,
        value: null
      });
    },
    handleConfirm: function handleConfirm(event) {
      this.$emit('confirm', {
        event: event,
        value: null
      });
    }
  }
};
exports.SchedulerRemoveDialogVue2 = SchedulerRemoveDialogVue2;
/**
 * @hidden
 */
var SchedulerRemoveDialog = SchedulerRemoveDialogVue2;
exports.SchedulerRemoveDialog = SchedulerRemoveDialog;
