// @progress/kendo-vue-scheduler/dist/npm/components/SchedulerActionButtons.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SchedulerActionButtonsVue2 = exports.SchedulerActionButtons = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_vue_dialogs_1 = require("@progress/kendo-vue-dialogs");
var kendo_svg_icons_1 = require("@progress/kendo-svg-icons");
/**
 * @hidden
 */
var SchedulerActionButtonsVue2 = {
  name: 'KendoSchedulerActionButtons',
  props: {
    editSaveMessage: String,
    editCancelMessage: String
  },
  inject: {
    kendoForm: {
      default: {}
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var _this = this;
    var h = gh || createElement;
    return (
      // @ts-ignore  function children
      h(kendo_vue_dialogs_1.DialogActionsBar, {
        "class": 'k-form-buttons',
        layout: 'start',
        attrs: this.v3 ? undefined : {
          layout: 'start'
        }
      }, this.v3 ? function () {
        return [
        // @ts-ignore  function children
        h(kendo_vue_buttons_1.Button, {
          type: "submit",
          attrs: _this.v3 ? undefined : {
            type: "submit",
            themeColor: 'primary',
            disabled: !_this.kendoForm.allowSubmit,
            icon: 'save',
            svgIcon: kendo_svg_icons_1.saveIcon
          },
          themeColor: 'primary',
          disabled: !_this.kendoForm.allowSubmit,
          icon: 'save',
          svgIcon: kendo_svg_icons_1.saveIcon,
          onClick: _this.onSubmit,
          on: _this.v3 ? undefined : {
            "click": _this.onSubmit
          }
        }, _this.v3 ? function () {
          return [_this.editSaveMessage];
        } : [_this.editSaveMessage]),
        // @ts-ignore  function children
        h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleCancel,
          on: _this.v3 ? undefined : {
            "click": _this.handleCancel
          },
          icon: 'cancel',
          attrs: _this.v3 ? undefined : {
            icon: 'cancel',
            svgIcon: kendo_svg_icons_1.cancelIcon
          },
          svgIcon: kendo_svg_icons_1.cancelIcon
        }, _this.v3 ? function () {
          return [_this.editCancelMessage];
        } : [_this.editCancelMessage])];
      } : [h(kendo_vue_buttons_1.Button, {
        type: "submit",
        attrs: _this.v3 ? undefined : {
          type: "submit",
          themeColor: 'primary',
          disabled: !_this.kendoForm.allowSubmit,
          icon: 'save',
          svgIcon: kendo_svg_icons_1.saveIcon
        },
        themeColor: 'primary',
        disabled: !_this.kendoForm.allowSubmit,
        icon: 'save',
        svgIcon: kendo_svg_icons_1.saveIcon,
        onClick: _this.onSubmit,
        on: _this.v3 ? undefined : {
          "click": _this.onSubmit
        }
      }, _this.v3 ? function () {
        return [_this.editSaveMessage];
      } : [_this.editSaveMessage]), h(kendo_vue_buttons_1.Button, {
        onClick: _this.handleCancel,
        on: _this.v3 ? undefined : {
          "click": _this.handleCancel
        },
        icon: 'cancel',
        attrs: _this.v3 ? undefined : {
          icon: 'cancel',
          svgIcon: kendo_svg_icons_1.cancelIcon
        },
        svgIcon: kendo_svg_icons_1.cancelIcon
      }, _this.v3 ? function () {
        return [_this.editCancelMessage];
      } : [_this.editCancelMessage])])
    );
  },
  methods: {
    onSubmit: function onSubmit(e) {
      this.kendoForm.onSubmit(e);
    },
    handleCancel: function handleCancel(e) {
      this.$emit('cancel', e);
    }
  }
};
exports.SchedulerActionButtonsVue2 = SchedulerActionButtonsVue2;
/**
 * @hidden
 */
var SchedulerActionButtons = SchedulerActionButtonsVue2;
exports.SchedulerActionButtons = SchedulerActionButtons;
