// @progress/kendo-vue-scheduler/dist/npm/components/SchedulerOccurrenceDialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SchedulerOccurrenceDialogVue2 = exports.SchedulerOccurrenceDialog = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var kendo_vue_dialogs_1 = require("@progress/kendo-vue-dialogs");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var main_1 = require("../messages/main");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
/**
 * @hidden
 */
var SchedulerOccurrenceDialogVue2 = {
  name: 'KendoSchedulerOccurrenceDialog',
  props: {
    dataItem: Object,
    isRemove: Boolean
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  created: function created() {
    this.dialogId = (0, kendo_vue_common_1.guid)();
  },
  mounted: function mounted() {
    var that = this;
    setTimeout(function () {
      var dialogWrapper = document.getElementById(that.dialogId);
      if (dialogWrapper) {
        dialogWrapper.focus();
      }
    }, 10);
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    var kendoLocalizationService = inject('kendoLocalizationService', {});
    return {
      v3: v3,
      kendoLocalizationService: kendoLocalizationService
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var _this = this;
    var h = gh || createElement;
    var isRemove = this.$props.isRemove;
    var localizationService = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    var localizeMessage = function localizeMessage(message) {
      return localizationService.toLanguageString(message, main_1.messages[message]);
    };
    var editOccurrenceMessage = localizeMessage(main_1.editorOccurrence);
    var editSeriesMessage = localizeMessage(main_1.editorSeries);
    var editRecurringConfirmationMessage = localizeMessage(main_1.editorRecurringConfirmation);
    var editRecurringDialogTitleMessage = localizeMessage(main_1.editorRecurringDialogTitle);
    var deleteOccurrenceMessage = localizeMessage(main_1.deleteOccurrence);
    var deleteSeriesMessage = localizeMessage(main_1.deleteSeries);
    var deleteRecurringDialogTitleMessage = localizeMessage(main_1.deleteRecurringDialogTitle);
    var deleteRecurringConfirmationMessage = localizeMessage(main_1.deleteRecurringConfirmation);
    var titleMessage = isRemove ? deleteRecurringDialogTitleMessage : editRecurringDialogTitleMessage;
    var confirmationMessage = isRemove ? deleteRecurringConfirmationMessage : editRecurringConfirmationMessage;
    var occurrenceMessage = isRemove ? deleteOccurrenceMessage : editOccurrenceMessage;
    var seriesMessage = isRemove ? deleteSeriesMessage : editSeriesMessage;
    return (
      // @ts-ignore  function children
      h(kendo_vue_dialogs_1.Dialog, {
        title: titleMessage,
        attrs: this.v3 ? undefined : {
          title: titleMessage,
          appendTo: 'body',
          wrapperId: this.dialogId
        },
        appendTo: 'body',
        wrapperId: this.dialogId,
        onClose: this.handleClose,
        on: this.v3 ? undefined : {
          "close": this.handleClose
        }
      }, this.v3 ? function () {
        return [confirmationMessage,
        // @ts-ignore  function children
        h(kendo_vue_dialogs_1.DialogActionsBar, _this.v3 ? function () {
          return [
          // @ts-ignore  function children
          h(kendo_vue_buttons_1.Button, {
            onClick: _this.handleOccurrenceClick,
            on: _this.v3 ? undefined : {
              "click": _this.handleOccurrenceClick
            }
          }, _this.v3 ? function () {
            return [occurrenceMessage];
          } : [occurrenceMessage]),
          // @ts-ignore  function children
          h(kendo_vue_buttons_1.Button, {
            onClick: _this.handleSeriesClick,
            on: _this.v3 ? undefined : {
              "click": _this.handleSeriesClick
            }
          }, _this.v3 ? function () {
            return [seriesMessage];
          } : [seriesMessage])];
        } : [h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleOccurrenceClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleOccurrenceClick
          }
        }, _this.v3 ? function () {
          return [occurrenceMessage];
        } : [occurrenceMessage]), h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleSeriesClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleSeriesClick
          }
        }, _this.v3 ? function () {
          return [seriesMessage];
        } : [seriesMessage])])];
      } : [confirmationMessage, h(kendo_vue_dialogs_1.DialogActionsBar, _this.v3 ? function () {
        return [h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleOccurrenceClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleOccurrenceClick
          }
        }, _this.v3 ? function () {
          return [occurrenceMessage];
        } : [occurrenceMessage]), h(kendo_vue_buttons_1.Button, {
          onClick: _this.handleSeriesClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleSeriesClick
          }
        }, _this.v3 ? function () {
          return [seriesMessage];
        } : [seriesMessage])];
      } : [h(kendo_vue_buttons_1.Button, {
        onClick: _this.handleOccurrenceClick,
        on: _this.v3 ? undefined : {
          "click": _this.handleOccurrenceClick
        }
      }, _this.v3 ? function () {
        return [occurrenceMessage];
      } : [occurrenceMessage]), h(kendo_vue_buttons_1.Button, {
        onClick: _this.handleSeriesClick,
        on: _this.v3 ? undefined : {
          "click": _this.handleSeriesClick
        }
      }, _this.v3 ? function () {
        return [seriesMessage];
      } : [seriesMessage])])])
    );
  },
  methods: {
    handleClose: function handleClose(_a) {
      var event = _a.event;
      this.$emit('close', {
        event: event,
        value: null
        // target: target.current
      });
    },

    handleSeriesClick: function handleSeriesClick(event) {
      this.$emit('seriesclose', {
        event: event,
        value: null
        // target: target.current 
      });
    },

    handleOccurrenceClick: function handleOccurrenceClick(event) {
      this.$emit('occurrenceclose', {
        event: event,
        value: null
        // target: target.current 
      });
    }
  }
};

exports.SchedulerOccurrenceDialogVue2 = SchedulerOccurrenceDialogVue2;
/**
 * @hidden
 */
var SchedulerOccurrenceDialog = SchedulerOccurrenceDialogVue2;
exports.SchedulerOccurrenceDialog = SchedulerOccurrenceDialog;
