// @progress/kendo-vue-scheduler/dist/npm/components/ShowMoreItemsButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ShowMoreItemsButtonVue2 = exports.ShowMoreItemsButton = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var kendo_svg_icons_1 = require("@progress/kendo-svg-icons");
/**
 * @hidden
 */
var ShowMoreItemsButtonVue2 = {
  name: 'KendoShowMoreItemsButton',
  props: {
    //  currentSlot: Object
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    var kendoLocalizationService = inject('kendoLocalizationService', {});
    return {
      v3: v3,
      kendoLocalizationService: kendoLocalizationService
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var _this = this;
    var h = gh || createElement;
    return (
      // @ts-ignore  function children
      h(kendo_vue_buttons_1.Button, {
        tabIndex: -1,
        attrs: this.v3 ? undefined : {
          tabIndex: -1
        },
        "class": "k-more-events",
        onClick: this.handleClick,
        on: this.v3 ? undefined : {
          "click": this.handleClick
        }
      }, this.v3 ? function () {
        return [h(kendo_vue_common_1.Icon, {
          name: "more-horizontal",
          attrs: _this.v3 ? undefined : {
            name: "more-horizontal",
            icon: kendo_svg_icons_1.moreHorizontalIcon
          },
          icon: kendo_svg_icons_1.moreHorizontalIcon
        })];
      } : [h(kendo_vue_common_1.Icon, {
        name: "more-horizontal",
        attrs: _this.v3 ? undefined : {
          name: "more-horizontal",
          icon: kendo_svg_icons_1.moreHorizontalIcon
        },
        icon: kendo_svg_icons_1.moreHorizontalIcon
      })])
    );
  },
  methods: {
    handleClick: function handleClick(event) {
      this.$emit('click', {
        target: {
          slot: this.currentSlot
        },
        event: event
      });
    }
  }
};
exports.ShowMoreItemsButtonVue2 = ShowMoreItemsButtonVue2;
/**
 * @hidden
 */
var ShowMoreItemsButton = ShowMoreItemsButtonVue2;
exports.ShowMoreItemsButton = ShowMoreItemsButton;
