// @progress/kendo-vue-editor/dist/esm/tools/insertImage.js
var __assign = this && this.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];
      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
var __rest = this && this.__rest || function (s, e) {
  var t = {};
  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }
  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
import { canInsert } from '@progress/kendo-editor-common';
import { Button as kbutton } from '@progress/kendo-vue-buttons';
import { InsertImageDialog } from '../dialogs/insertImage.js';
import { provideLocalizationService } from '@progress/kendo-vue-intl';
import { messages } from '../messages/main.js';
/**
 * @hidden
 */
var InsertImageVue2 = {
  name: 'KendoInsertImage',
  props: __assign({
    view: Object,
    settings: {
      type: Object,
      default: function _default() {
        return {
          messages: {}
        };
      }
    },
    dir: String
  }, kbutton.props),
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  data: function data() {
    return {
      openedDialog: false
    };
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
      view = _a.view,
      settings = _a.settings,
      other = __rest(_a, ["view", "settings"]);
    var state = view && view.state;
    var nodeType = state ? state.schema.nodes[settings.node] : undefined;
    var localization = provideLocalizationService(this);
    var insertImage = settings.messages.insertImage;
    var buttonProps = __assign(__assign(__assign({}, other), {
      key: 'insertImage',
      disabled: !nodeType || !state || !canInsert(state, nodeType),
      title: localization.toLanguageString(insertImage, messages[insertImage])
    }), settings.props);
    var button = h(kbutton, __assign({
      onClick: this.toggleDialog,
      onMousedown: this.preventDefault,
      onPointerdown: this.preventDefault,
      on: this.v3 ? undefined : {
        'click': this.toggleDialog,
        'mousedown': this.preventDefault,
        'pointerdown': this.preventDefault
      },
      attrs: this.v3 ? undefined : buttonProps
    }, buttonProps));
    var rendering = [button, this.openedDialog && view &&
    // @ts-ignore function children
    h(InsertImageDialog, {
      key: "insertImageDialog",
      view: view,
      attrs: this.v3 ? undefined : {
        view: view,
        settings: settings,
        dir: buttonProps.dir
      },
      settings: settings,
      dir: buttonProps.dir,
      onClose: this.toggleDialog,
      on: this.v3 ? undefined : {
        "close": this.toggleDialog
      }
    }) || null];
    return h("span", [rendering]);
  },
  methods: {
    toggleDialog: function toggleDialog() {
      this.openedDialog = !this.openedDialog;
    },
    preventDefault: function preventDefault(e) {
      e.preventDefault();
    }
  }
};
/**
 * @hidden
 */
var InsertImage = InsertImageVue2;
export { InsertImage, InsertImageVue2 };
