// @progress/kendo-vue-scheduler/dist/npm/items/SchedulerItemContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SchedulerItemContentVue2 = exports.SchedulerItemContent = void 0;
// @ts-ignore
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
/**
 * @hidden
 */
var SchedulerItemContentVue2 = {
  name: 'KendoSchedulerItemContent',
  props: {
    title: String
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlot = (0, kendo_vue_common_1.getDefaultSlots)(this);
    return h("div", {
      "class": 'k-event-template'
    }, [this.title, defaultSlot]);
  }
};
exports.SchedulerItemContentVue2 = SchedulerItemContentVue2;
/**
 * @hidden
 */
var SchedulerItemContent = SchedulerItemContentVue2;
exports.SchedulerItemContent = SchedulerItemContent;
