// @progress/kendo-vue-form/dist/es/FieldWrapper.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
import { packageMetadata } from './package-metadata';
import { getDefaultSlots, validatePackage } from '@progress/kendo-vue-common';
/**
 * @hidden
 */
var FieldWrapperVue2 = {
  name: 'KendoFieldWrapper',
  props: {
    dir: String
  },
  created: function created() {
    validatePackage(packageMetadata);
  },
  computed: {
    fieldClassName: function fieldClassName() {
      return {
        'k-form-field': true,
        'k-rtl': this.$props.dir === 'rtl'
      };
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlots = getDefaultSlots(this);
    return h("div", {
      "class": this.fieldClassName
    }, [defaultSlots]);
  }
};
/**
 * @hidden
 */
var FieldWrapper = FieldWrapperVue2;
export { FieldWrapper, FieldWrapperVue2 };
