// @progress/kendo-vue-scheduler/dist/npm/views/day/DayViewRowContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DayViewRowContentVue2 = exports.DayViewRowContent = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var TimeHeaderCell_1 = require("../../components/TimeHeaderCell");
/** @hidden */
var DayViewRowContentVue2 = {
  name: 'KendoDayViewRowContent',
  props: {
    contentSlot: Object,
    // slot: Object,
    isMaster: Boolean,
    isLast: Boolean,
    timeHeaderCell: [String, Function, Object]
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var contentSlot = this.$props.contentSlot;
    return this.$props.isMaster ? h(TimeHeaderCell_1.TimeHeaderCell, {
      as: this.$props.timeHeaderCell,
      attrs: this.v3 ? undefined : {
        as: this.$props.timeHeaderCell,
        date: contentSlot.zonedStart,
        start: contentSlot.zonedStart,
        end: contentSlot.zonedEnd
      },
      date: contentSlot.zonedStart,
      start: contentSlot.zonedStart,
      end: contentSlot.zonedEnd,
      "class": (0, kendo_vue_common_1.classNames)('k-side-cell', {
        'k-major-cell': !this.$props.isLast
      })
    }) : h("div", {
      "class": 'k-scheduler-cell k-heading-cell k-side-cell'
    });
  }
};
exports.DayViewRowContentVue2 = DayViewRowContentVue2;
/** @hidden */
var DayViewRowContent = DayViewRowContentVue2;
exports.DayViewRowContent = DayViewRowContent;
