// @progress/kendo-vue-scheduler/dist/npm/views/day/DayViewHead.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DayViewHeadVue2 = exports.DayViewHead = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var DateHeaderCell_1 = require("../../components/DateHeaderCell");
var kendo_date_math_1 = require("@progress/kendo-date-math");
/** @hidden */
var DayViewHeadVue2 = {
  name: 'KendoDayViewHead',
  props: {
    timeRanges: Array,
    dateHeaderCell: [String, Function, Object]
  },
  inject: {
    ri: {
      default: null
    },
    ks: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    return h("div", {
      "class": "k-scheduler-row",
      key: this.ri.groupIndex
    }, [this.timeRanges.map(function (range, rangeIndex) {
      return h(DateHeaderCell_1.DateHeaderCell, {
        as: this.$props.dateHeaderCell,
        attrs: this.v3 ? undefined : {
          as: this.$props.dateHeaderCell,
          date: kendo_date_math_1.ZonedDate.fromLocalDate(new Date(range.zonedEnd.getTime() - (range.zonedEnd.getTime() - range.zonedStart.getTime()) / 2), this.ks.timezone),
          start: range.start,
          end: range.end,
          format: {
            skeleton: 'MEd'
          }
        },
        key: rangeIndex,
        date: kendo_date_math_1.ZonedDate.fromLocalDate(new Date(range.zonedEnd.getTime() - (range.zonedEnd.getTime() - range.zonedStart.getTime()) / 2), this.ks.timezone),
        start: range.start,
        end: range.end,
        format: {
          skeleton: 'MEd'
        }
      });
    }, this)]);
  }
};
exports.DayViewHeadVue2 = DayViewHeadVue2;
/** @hidden */
var DayViewHead = DayViewHeadVue2;
exports.DayViewHead = DayViewHead;
