// @progress/kendo-vue-scheduler/dist/npm/views/day/DayView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DayViewVue2 = exports.DayView = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var MultiDayView_1 = require("../day/MultiDayView");
var main_1 = require("../../messages/main");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var kendo_date_math_1 = require("@progress/kendo-date-math");
/** @hidden */
var DayViewVue2 = {
  name: 'KendoDayView',
  // @ts-ignore
  emits: {
    dataaction: null
  },
  props: {
    name: {
      type: String,
      default: 'day'
    },
    title: String,
    dateRange: {
      type: [Function, Object],
      default: function _default() {
        return MultiDayView_1.MultiDayView.props.dateRange.default();
      }
    },
    showWorkHours: {
      type: Boolean,
      default: true
    },
    step: {
      type: Number,
      default: 1
    },
    numberOfDays: {
      type: Number,
      default: 1
    },
    slotDuration: {
      type: Number,
      default: 60
    },
    slotDivisions: {
      type: Number,
      default: 2
    },
    selectedDateFormat: {
      type: String,
      default: '{0:D}'
    },
    selectedShortDateFormat: {
      type: String,
      default: '{0:d}'
    },
    currentTimeMarker: {
      type: Boolean,
      default: true
    },
    startTime: {
      type: String,
      default: '00:00'
    },
    endTime: {
      type: String,
      default: '23:59'
    },
    workDayStart: {
      type: String,
      default: '08:00'
    },
    workDayEnd: {
      type: String,
      default: '17:00'
    },
    workWeekStart: {
      type: Number,
      default: kendo_date_math_1.Day.Monday
    },
    workWeekEnd: {
      type: Number,
      default: kendo_date_math_1.Day.Friday
    },
    defaultShowWorkHours: {
      type: Boolean,
      default: true
    },
    dateHeaderCell: [String, Function, Object],
    timeHeaderCell: [String, Function, Object],
    slotRender: [String, Function, Object],
    viewSlot: [String, Function, Object],
    resizeHint: [String, Function, Object],
    item: [String, Function, Object],
    viewItem: [String, Function, Object],
    editable: {
      type: [Boolean, Object],
      default: false
    },
    format: String
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    var title = localization.toLanguageString(main_1.dayViewTitle, main_1.messages[main_1.dayViewTitle]);
    return h(MultiDayView_1.MultiDayView, {
      name: this.$props.name,
      attrs: this.v3 ? undefined : {
        name: this.$props.name,
        dateRange: this.$props.dateRange,
        step: this.$props.step,
        numberOfDays: this.$props.numberOfDays,
        slotDuration: this.$props.slotDuration,
        slotDivisions: this.$props.slotDivisions,
        selectedDateFormat: this.$props.selectedDateFormat,
        selectedShortDateFormat: this.$props.selectedShortDateFormat,
        showWorkHours: this.$props.showWorkHours,
        title: title,
        currentTimeMarker: this.$props.currentTimeMarker,
        startTime: this.$props.startTime,
        endTime: this.$props.endTime,
        workDayStart: this.$props.workDayStart,
        workDayEnd: this.$props.workDayEnd,
        workWeekStart: this.$props.workWeekStart,
        workWeekEnd: this.$props.workWeekEnd,
        defaultShowWorkHours: this.$props.defaultShowWorkHours,
        timeHeaderCell: this.$props.timeHeaderCell,
        dateHeaderCell: this.$props.dateHeaderCell,
        slotRender: this.$props.slotRender,
        viewSlot: this.$props.viewSlot,
        resizeHint: this.$props.resizeHint,
        item: this.$props.item,
        viewItem: this.$props.viewItem,
        slotItem: this.$props.slotItem,
        editable: this.$props.editable
      },
      dateRange: this.$props.dateRange,
      step: this.$props.step,
      numberOfDays: this.$props.numberOfDays,
      slotDuration: this.$props.slotDuration,
      slotDivisions: this.$props.slotDivisions,
      selectedDateFormat: this.$props.selectedDateFormat,
      selectedShortDateFormat: this.$props.selectedShortDateFormat,
      showWorkHours: this.$props.showWorkHours,
      title: title,
      currentTimeMarker: this.$props.currentTimeMarker,
      startTime: this.$props.startTime,
      endTime: this.$props.endTime,
      workDayStart: this.$props.workDayStart,
      workDayEnd: this.$props.workDayEnd,
      workWeekStart: this.$props.workWeekStart,
      workWeekEnd: this.$props.workWeekEnd,
      defaultShowWorkHours: this.$props.defaultShowWorkHours,
      timeHeaderCell: this.$props.timeHeaderCell,
      dateHeaderCell: this.$props.dateHeaderCell,
      slotRender: this.$props.slotRender,
      viewSlot: this.$props.viewSlot,
      resizeHint: this.$props.resizeHint,
      item: this.$props.item,
      viewItem: this.$props.viewItem,
      slotItem: this.$props.slotItem,
      editable: this.$props.editable,
      onDataaction: this.handleDataAction,
      on: this.v3 ? undefined : {
        "dataaction": this.handleDataAction
      }
    });
  },
  methods: {
    handleDataAction: function handleDataAction(action) {
      this.$emit('dataaction', action);
    }
  }
};
exports.DayViewVue2 = DayViewVue2;
/** @hidden */
var DayView = DayViewVue2;
exports.DayView = DayView;
