// @progress/kendo-vue-scheduler/dist/npm/views/day/DayViewAllDayRowContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DayViewAllDayRowContentVue2 = exports.DayViewAllDayRowContent = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var main_1 = require("../../messages/main");
/** @hidden */
var DayViewAllDayRowContentVue2 = {
  name: 'KendoDayViewAllDayRowContent',
  props: {
    groupIndex: Number
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    return (
      // fragment
      h("div", {
        "class": "k-scheduler-cell k-heading-cell k-side-cell k-scheduler-times-all-day"
      }, [localization.toLanguageString(main_1.allDay, main_1.messages[main_1.allDay])])
    );
  }
};
exports.DayViewAllDayRowContentVue2 = DayViewAllDayRowContentVue2;
/** @hidden */
var DayViewAllDayRowContent = DayViewAllDayRowContentVue2;
exports.DayViewAllDayRowContent = DayViewAllDayRowContent;
