// @progress/kendo-vue-scheduler/dist/npm/views/day/DayViewGroupRowContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DayViewGroupRowContentVue2 = exports.DayViewGroupRowContent = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
/** @hidden */
var DayViewGroupRowContentVue2 = {
  name: 'KendoDayViewGroupRowContent',
  props: {
    rangeIndex: Number
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    return h("div", {
      "class": "k-scheduler-cell k-side-cell",
      "data-range-index": this.$props.rangeIndex,
      attrs: this.v3 ? undefined : {
        "data-range-index": this.$props.rangeIndex
      }
    });
  }
};
exports.DayViewGroupRowContentVue2 = DayViewGroupRowContentVue2;
/** @hidden */
var DayViewGroupRowContent = DayViewGroupRowContentVue2;
exports.DayViewGroupRowContent = DayViewGroupRowContent;
