// @progress/kendo-vue-scheduler/dist/es/components/header/navigation/DatePickerToggleButton.js
var __assign = this && this.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];
      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
import { provideIntlService } from '@progress/kendo-vue-intl';
import { MS_PER_DAY } from '../../../constants/main';
import { Button } from '@progress/kendo-vue-buttons';
import { calendarIcon } from '@progress/kendo-svg-icons';
/**
 * @hidden
 */
var DatePickerToggleButtonVue2 = {
  name: 'KendoDatePickerToggleButton',
  props: __assign({}, Button.props),
  inheritAttrs: false,
  inject: {
    kendoIntlService: {
      default: null
    },
    ks: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var intl = provideIntlService(this);
    // const { dateFormat, shortDateFormat } = useSchedulerDateFormatContext();
    // const dateRange = useSchedulerDateRangeContext();
    // const [date] = useSchedulerDateContext();
    var isMonthView = this.ks.dateRange.end.getTime() - this.ks.dateRange.start.getTime() > MS_PER_DAY * 27;
    var text = intl.format(this.ks.dateFormat, isMonthView ? this.ks.currentDate : this.ks.dateRange.zonedStart, isMonthView ? this.ks.currentDate : this.ks.dateRange.zonedEnd.addDays(-1));
    var shortText = intl.format(this.ks.shortDateFormat, isMonthView ? this.ks.currentDate : this.ks.dateRange.zonedStart, isMonthView ? this.ks.currentDate : this.ks.dateRange.zonedEnd.addDays(-1));
    return (
      // @ts-ignore  function children
      h(Button, {
        onClick: this.handleClick
        // {...props} 
        ,
        on: this.v3 ? undefined : {
          "click": this.handleClick
        },
        icon: "calendar",
        attrs: this.v3 ? undefined : {
          icon: "calendar",
          svgIcon: calendarIcon,
          tabIndex: -1,
          fillMode: 'flat'
        },
        svgIcon: calendarIcon,
        "class": "k-nav-current",
        tabIndex: -1,
        fillMode: 'flat'
      }, this.v3 ? function () {
        return [h("span", {
          "class": "k-button-text"
        }, [h("span", {
          "class": "k-sm-date-format"
        }, [shortText]), h("span", {
          "class": "k-lg-date-format"
        }, [text])])];
      } : [h("span", {
        "class": "k-button-text"
      }, [h("span", {
        "class": "k-sm-date-format"
      }, [shortText]), h("span", {
        "class": "k-lg-date-format"
      }, [text])])])
    );
  },
  methods: {
    handleClick: function handleClick(e) {
      this.$emit('click', e);
    }
  }
};
/**
 * @hidden
 */
var DatePickerToggleButton = DatePickerToggleButtonVue2;
export { DatePickerToggleButton, DatePickerToggleButtonVue2 };
