// @progress/kendo-vue-scheduler/dist/npm/components/header/SchedulerHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SchedulerHeaderVue2 = exports.SchedulerHeader = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
/**
 * @hidden
 */
var SchedulerHeaderVue2 = {
  name: 'KendoSchedulerHeader',
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlot = (0, kendo_vue_common_1.getDefaultSlots)(this);
    return (
      // @ts-ignore  function children
      h(kendo_vue_buttons_1.Toolbar, {
        "class": 'k-scheduler-toolbar'
      }, this.v3 ? function () {
        return [defaultSlot];
      } : [defaultSlot])
    );
  }
};
exports.SchedulerHeaderVue2 = SchedulerHeaderVue2;
var SchedulerHeader = SchedulerHeaderVue2;
exports.SchedulerHeader = SchedulerHeader;
