// @progress/kendo-vue-scheduler/dist/es/views/agenda/AgendaViewHeaderItem.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
import { provideIntlService } from '@progress/kendo-vue-intl';
/**
 * @hidden
 */
var AgendaViewHeaderItemVue2 = {
  name: 'KendoAgendaViewHeaderItem',
  props: {
    range: Object
  },
  inject: {
    kendoIntlService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var intl = provideIntlService(this);
    return h("div", {
      "class": "k-scheduler-cell k-scheduler-datecolumn k-group-cell"
    }, [h("div", [h("strong", {
      "class": "k-scheduler-agendaday"
    }, [intl.formatDate(this.$props.range.zonedStart, 'dd')]), h("em", {
      "class": "k-scheduler-agendaweek"
    }, [intl.formatDate(this.$props.range.zonedStart, 'EEEE')]), h("span", {
      "class": "k-scheduler-agendadate"
    }, [intl.formatDate(this.$props.range.zonedStart, 'y')])])]);
  }
};
/**
 * @hidden
 */
var AgendaViewHeaderItem = AgendaViewHeaderItemVue2;
export { AgendaViewHeaderItem, AgendaViewHeaderItemVue2 };
