// @progress/kendo-vue-editor/dist/es/tools/findReplace.js
var __assign = this && this.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];
      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
var __rest = this && this.__rest || function (s, e) {
  var t = {};
  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }
  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
import { Button as kbutton } from '@progress/kendo-vue-buttons';
import { FindAndReplaceDialog } from '../dialogs/FindReplace';
import { messages } from '../messages/main';
import { EditorToolsSettings } from './../config/toolsSettings';
import { provideLocalizationService } from '@progress/kendo-vue-intl';
var settings = EditorToolsSettings.findAndReplace;
/**
 * @hidden
 */
var FindAndReplaceVue2 = {
  name: 'KendoFindAndReplace',
  props: __assign({
    view: Object
  }, kbutton.props),
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  data: function data() {
    return {
      showDialog: false
    };
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
      view = _a.view,
      other = __rest(_a, ["view"]);
    var localization = provideLocalizationService(this);
    var titleKey = settings.messages.findReplaceToolTitle;
    var buttonProps = __assign(__assign(__assign({}, other), {
      key: 'tool',
      disabled: !view,
      title: localization.toLanguageString(titleKey, messages[titleKey])
    }), settings.props);
    var button = h(kbutton, __assign({
      onClick: this.onOpen,
      onMousedown: this.preventDefault,
      onPointerdown: this.preventDefault,
      on: this.v3 ? undefined : {
        'click': this.onOpen,
        'mousedown': this.preventDefault,
        'pointerdown': this.preventDefault
      },
      attrs: this.v3 ? undefined : buttonProps
    }, buttonProps));
    return h("span", [button, this.showDialog && view &&
    // @ts-ignore function children
    h(FindAndReplaceDialog, {
      view: view,
      attrs: this.v3 ? undefined : {
        view: view,
        dir: this.$props.dir
      },
      onClose: this.onClose,
      on: this.v3 ? undefined : {
        "close": this.onClose
      },
      dir: this.$props.dir,
      key: "dialog"
    })]);
  },
  methods: {
    onClose: function onClose() {
      this.showDialog = false;
    },
    onOpen: function onOpen() {
      this.showDialog = true;
    },
    preventDefault: function preventDefault(e) {
      e.preventDefault();
    }
  }
};
/**
 * @hidden
 */
var FindAndReplace = FindAndReplaceVue2;
export { FindAndReplace, FindAndReplaceVue2 };
