// @progress/kendo-vue-listview/dist/npm/ListView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListViewVue2 = exports.ListView = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var package_metadata_1 = require("./package-metadata");
/**
 * @hidden
 */
var ListViewVue2 = {
  name: 'KendoListView',
  // @ts-ignore
  emits: {
    scroll: null
  },
  props: {
    dataItems: Array,
    item: [String, Function, Object],
    header: [String, Function, Object],
    footer: [String, Function, Object]
  },
  created: function created() {
    (0, kendo_vue_common_1.validatePackage)(package_metadata_1.packageMetadata);
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
      dataItems = _a.dataItems,
      item = _a.item,
      header = _a.header,
      footer = _a.footer;
    var headerTemplate = kendo_vue_common_1.templateRendering.call(this, header, kendo_vue_common_1.getListeners.call(this));
    var footerTemplate = kendo_vue_common_1.templateRendering.call(this, footer, kendo_vue_common_1.getListeners.call(this));
    var headerRender = kendo_vue_common_1.getTemplate.call(this, {
      h: h,
      template: headerTemplate
    });
    var footerRender = kendo_vue_common_1.getTemplate.call(this, {
      h: h,
      template: footerTemplate
    });
    return h("div", {
      "class": 'k-listview k-listview-bordered k-d-flex'
    }, [header && h("div", {
      "class": 'k-listview-header'
    }, [headerRender]), h("div", {
      "class": "k-listview-content",
      role: 'list',
      attrs: this.v3 ? undefined : {
        role: 'list'
      },
      onScroll: this.onScroll,
      on: this.v3 ? undefined : {
        "scroll": this.onScroll
      }
    }, [item && dataItems.map(function (dataItem, index) {
      var itemTemplate = kendo_vue_common_1.templateRendering.call(this, item, kendo_vue_common_1.getListeners.call(this));
      var itemRender = kendo_vue_common_1.getTemplate.call(this, {
        h: h,
        template: itemTemplate,
        additionalProps: {
          dataItem: dataItem,
          index: index
        }
      });
      return h("div", {
        "class": 'k-listview-item',
        role: 'listitem',
        attrs: this.v3 ? undefined : {
          role: 'listitem',
          tabIndex: index === 0 ? 0 : undefined,
          "aria-posinset": index + 1,
          "aria-setsize": dataItems.length
        },
        tabIndex: index === 0 ? 0 : undefined,
        "aria-posinset": index + 1,
        "aria-setsize": dataItems.length
      }, [itemRender]);
    }, this)]), footer && h("div", {
      "class": 'k-listview-footer'
    }, [footerRender])]);
  },
  methods: {
    onScroll: function onScroll(event) {
      this.$emit('scroll', event, this);
    }
  }
};
exports.ListViewVue2 = ListViewVue2;
/**
 * @hidden
 */
var ListView = ListViewVue2;
exports.ListView = ListView;
