// @progress/kendo-vue-scheduler/dist/esm/components/footer/SchedulerFooter.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
import { getDefaultSlots } from '@progress/kendo-vue-common';
import { Toolbar } from '@progress/kendo-vue-buttons';
/**
 * @hidden
 */
var SchedulerFooterVue2 = {
  name: 'KendoSchedulerFooter',
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlot = getDefaultSlots(this);
    return (
      // @ts-ignore  function children
      h(Toolbar
      //   ref={(toolbar: Toolbar) => { if (toolbar) { element.current = toolbar.element; } }}
      , {
        "class": 'k-scheduler-footer',
        style: {
          boxShadow: 'none'
        }
      }, this.v3 ? function () {
        return [defaultSlot];
      } : [defaultSlot])
    );
  }
};
/**
 * @hidden
 */
var SchedulerFooter = SchedulerFooterVue2;
export { SchedulerFooter, SchedulerFooterVue2 };
