// @progress/kendo-vue-listview/dist/es/ListView.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
import { getTemplate, validatePackage, templateRendering, getListeners } from '@progress/kendo-vue-common';
import { packageMetadata } from './package-metadata';
/**
 * @hidden
 */
var ListViewVue2 = {
  name: 'KendoListView',
  // @ts-ignore
  emits: {
    scroll: null
  },
  props: {
    dataItems: Array,
    item: [String, Function, Object],
    header: [String, Function, Object],
    footer: [String, Function, Object]
  },
  created: function created() {
    validatePackage(packageMetadata);
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
      dataItems = _a.dataItems,
      item = _a.item,
      header = _a.header,
      footer = _a.footer;
    var headerTemplate = templateRendering.call(this, header, getListeners.call(this));
    var footerTemplate = templateRendering.call(this, footer, getListeners.call(this));
    var headerRender = getTemplate.call(this, {
      h: h,
      template: headerTemplate
    });
    var footerRender = getTemplate.call(this, {
      h: h,
      template: footerTemplate
    });
    return h("div", {
      "class": 'k-listview k-listview-bordered k-d-flex'
    }, [header && h("div", {
      "class": 'k-listview-header'
    }, [headerRender]), h("div", {
      "class": "k-listview-content",
      role: 'list',
      attrs: this.v3 ? undefined : {
        role: 'list'
      },
      onScroll: this.onScroll,
      on: this.v3 ? undefined : {
        "scroll": this.onScroll
      }
    }, [item && dataItems.map(function (dataItem, index) {
      var itemTemplate = templateRendering.call(this, item, getListeners.call(this));
      var itemRender = getTemplate.call(this, {
        h: h,
        template: itemTemplate,
        additionalProps: {
          dataItem: dataItem,
          index: index
        }
      });
      return h("div", {
        "class": 'k-listview-item',
        role: 'listitem',
        attrs: this.v3 ? undefined : {
          role: 'listitem',
          tabIndex: index === 0 ? 0 : undefined,
          "aria-posinset": index + 1,
          "aria-setsize": dataItems.length
        },
        tabIndex: index === 0 ? 0 : undefined,
        "aria-posinset": index + 1,
        "aria-setsize": dataItems.length
      }, [itemRender]);
    }, this)]), footer && h("div", {
      "class": 'k-listview-footer'
    }, [footerRender])]);
  },
  methods: {
    onScroll: function onScroll(event) {
      this.$emit('scroll', event, this);
    }
  }
};
/**
 * @hidden
 */
var ListView = ListViewVue2;
export { ListView, ListViewVue2 };
