// @progress/kendo-vue-scheduler/dist/npm/hooks/use-slots-focus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.findDownSlot = exports.findRightSlot = exports.findLeftSlot = exports.findUpSlot = exports.dispatchFocusedSlots = exports.SLOTS_FOCUS_ACTION = void 0;
/**
 * Represents the available `action` types for the `SchedulerSlotsFocusAction` objects.
 *
 * The available actions are:
 * - `SLOT_FOCUS_UP`
 * - `SLOT_FOCUS_DOWN`
 * - `SLOT_FOCUS_LEFT`
 * - `SLOT_FOCUS_RIGHT`
 */
var SLOTS_FOCUS_ACTION;
(function (SLOTS_FOCUS_ACTION) {
  SLOTS_FOCUS_ACTION["up"] = "SLOT_FOCUS_UP";
  SLOTS_FOCUS_ACTION["down"] = "SLOT_FOCUS_DOWN";
  SLOTS_FOCUS_ACTION["left"] = "SLOT_FOCUS_LEFT";
  SLOTS_FOCUS_ACTION["right"] = "SLOT_FOCUS_RIGHT";
})(SLOTS_FOCUS_ACTION = exports.SLOTS_FOCUS_ACTION || (exports.SLOTS_FOCUS_ACTION = {}));
/** @hidden */
var dispatchFocusedSlots = function dispatchFocusedSlots(action, event) {
  switch (action.type) {
    case SLOTS_FOCUS_ACTION.up:
      {
        if (!action.slot || !this.bv.slots) {
          return;
        }
        var next = (0, exports.findUpSlot)(action.slot, this.bv.slots);
        if (next && next.$el) {
          event.preventDefault();
          next.$el.focus();
        }
        break;
      }
    case SLOTS_FOCUS_ACTION.down:
      {
        if (!action.slot || !this.bv.slots) {
          return;
        }
        var next = (0, exports.findDownSlot)(action.slot, this.bv.slots);
        if (next && next.$el) {
          event.preventDefault();
          next.$el.focus();
        }
        break;
      }
    case SLOTS_FOCUS_ACTION.left:
      {
        if (!action.slot || !this.bv.slots) {
          return;
        }
        var next = (0, exports.findLeftSlot)(action.slot, this.bv.slots);
        if (next && next.$el) {
          event.preventDefault();
          next.$el.focus();
        }
        break;
      }
    case SLOTS_FOCUS_ACTION.right:
      {
        if (!action.slot || !this.bv.slots) {
          return;
        }
        var next = (0, exports.findRightSlot)(action.slot, this.bv.slots);
        if (next && next.$el) {
          event.preventDefault();
          next.$el.focus();
        }
        break;
      }
    default:
      break;
  }
};
exports.dispatchFocusedSlots = dispatchFocusedSlots;
/** @hidden */
var findUpSlot = function findUpSlot(target, all) {
  return all.find(function (slot) {
    return slot !== null && target !== null && slot.cSlot.row === target.$props.row - 1 && slot.cSlot.col === target.$props.col;
  });
};
exports.findUpSlot = findUpSlot;
/** @hidden */
var findLeftSlot = function findLeftSlot(target, all) {
  return all.find(function (slot) {
    return slot !== null && target !== null && slot.cSlot.row === target.$props.row && slot.cSlot.col === target.$props.col - 1;
  });
};
exports.findLeftSlot = findLeftSlot;
/** @hidden */
var findRightSlot = function findRightSlot(target, all) {
  return all.find(function (slot) {
    return slot !== null && target !== null && slot.cSlot.row === target.$props.row && slot.cSlot.col === target.$props.col + 1;
  });
};
exports.findRightSlot = findRightSlot;
/** @hidden */
var findDownSlot = function findDownSlot(target, all) {
  return all.find(function (slot) {
    return slot !== null && target !== null && slot.cSlot.row === target.$props.row + 1 && slot.cSlot.col === target.$props.col;
  });
};
exports.findDownSlot = findDownSlot;
