// @progress/kendo-vue-editor/dist/esm/dialogs/main.js
// import { InsertLinkDialog as LinkDialog, InsertLinkDialogProps as LinkDialogProps } from './insertLink.js';
// import { InsertImageDialog as ImageDialog, InsertImageDialogProps as ImageDialogProps } from './insertImage.js';
// import { ViewHtmlDialog as ViewHtmlDialogX, ViewHtmlDialogProps as ViewHtmlDialogPropsX } from './viewHtml.js';
// import { FindAndReplaceDialog as FindAndReplaceDialogX,
// FindAndReplaceDialogProps as FindAndReplaceDialogPropsX } from './FindReplace.js';
// /**
//  * Represents a wrapping namespace for the dialog components and props of the Editor.
//  */
// export namespace EditorDialogs {
//     /**
//      * The InsertLinkDialog component.
//      */
//     export class InsertLinkDialog extends LinkDialog { }
//     /**
//      * The props for the InsertLinkDialog component.
//      */
//     export interface InsertLinkDialogProps extends LinkDialogProps { }
//     /**
//      * The InsertImageDialog component.
//      */
//     export class InsertImageDialog extends ImageDialog { }
//     /**
//      * The props for the InsertImageDialog component.
//      */
//     export interface InsertImageDialogProps extends ImageDialogProps { }
//     /**
//      * The ViewHtmlDialog component.
//      */
//     export class ViewHtmlDialog extends ViewHtmlDialogX { }
//     /**
//      * The props for the ViewHtmlDialog component.
//      */
//     export interface ViewHtmlDialogProps extends ViewHtmlDialogPropsX { }
//     /**
//      * The FindAndReplaceDialog component.
//      */
//     export class FindAndReplaceDialog extends FindAndReplaceDialogX { }
//     /**
//      * The props for the FindAndReplaceDialog component.
//      */
//     export interface FindAndReplaceDialogProps extends FindAndReplaceDialogPropsX { }
// }
