// @progress/kendo-vue-form/dist/npm/FieldWrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FieldWrapperVue2 = exports.FieldWrapper = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var package_metadata_1 = require("./package-metadata");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
/**
 * @hidden
 */
var FieldWrapperVue2 = {
  name: 'KendoFieldWrapper',
  props: {
    dir: String
  },
  created: function created() {
    (0, kendo_vue_common_1.validatePackage)(package_metadata_1.packageMetadata);
  },
  computed: {
    fieldClassName: function fieldClassName() {
      return {
        'k-form-field': true,
        'k-rtl': this.$props.dir === 'rtl'
      };
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlots = (0, kendo_vue_common_1.getDefaultSlots)(this);
    return h("div", {
      "class": this.fieldClassName
    }, [defaultSlots]);
  }
};
exports.FieldWrapperVue2 = FieldWrapperVue2;
/**
 * @hidden
 */
var FieldWrapper = FieldWrapperVue2;
exports.FieldWrapper = FieldWrapper;
