// @progress/kendo-vue-editor/dist/npm/Editor.js
"use strict";

var __assign = undefined && undefined.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];
      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
var __spreadArray = undefined && undefined.__spreadArray || function (to, from, pack) {
  if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
    if (ar || !(i in from)) {
      if (!ar) ar = Array.prototype.slice.call(from, 0, i);
      ar[i] = from[i];
    }
  }
  return to.concat(ar || Array.prototype.slice.call(from));
};
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EditorVue2 = exports.Editor = exports.allTools = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var markRaw = allVue.markRaw;
var toRaw = allVue.toRaw;
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var kendo_editor_common_1 = require("@progress/kendo-editor-common");
var schema_1 = require("./config/schema");
var defaultStyles_1 = require("./config/defaultStyles");
var toolsSettings_1 = require("./config/toolsSettings");
var insertLink_1 = require("./dialogs/insertLink");
var main_1 = require("./utils/main");
var props_key_1 = require("./utils/props-key");
var controlled_value_1 = require("./utils/controlled-value");
var link = toolsSettings_1.EditorToolsSettings.link,
  bold = toolsSettings_1.EditorToolsSettings.bold,
  italic = toolsSettings_1.EditorToolsSettings.italic,
  underline = toolsSettings_1.EditorToolsSettings.underline;
var browser_detection_1 = require("./utils/browser-detection");
var package_metadata_1 = require("./package-metadata");
var align_1 = require("./tools/align");
var indent_1 = require("./tools/indent");
var lists_1 = require("./tools/lists");
var outdent_1 = require("./tools/outdent");
var inlineFormat_1 = require("./tools/inlineFormat");
var fontStyle_1 = require("./tools/fontStyle");
var formatBlock_1 = require("./tools/formatBlock");
var proseMirrorTool_1 = require("./tools/proseMirrorTool");
var insertLink_2 = require("./tools/insertLink");
var unlink_1 = require("./tools/unlink");
var cleanFormatting_1 = require("./tools/cleanFormatting");
var selectAll_1 = require("./tools/selectAll");
var insertImage_1 = require("./tools/insertImage");
var tool_1 = require("./tools/insertTable/tool");
var viewHtml_1 = require("./tools/viewHtml");
var pdf_1 = require("./tools/pdf");
var print_1 = require("./tools/print");
var findReplace_1 = require("./tools/findReplace");
var applyColor_1 = require("./tools/applyColor");
var main_2 = require("./messages/main");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
/** @hidden */
exports.allTools = {
  'Bold': {
    comp: inlineFormat_1.InlineFormat,
    props: toolsSettings_1.EditorToolsSettings.bold
  },
  'Italic': {
    comp: inlineFormat_1.InlineFormat,
    props: toolsSettings_1.EditorToolsSettings.italic
  },
  'Underline': {
    comp: inlineFormat_1.InlineFormat,
    props: toolsSettings_1.EditorToolsSettings.underline
  },
  'Strikethrough': {
    comp: inlineFormat_1.InlineFormat,
    props: toolsSettings_1.EditorToolsSettings.strikethrough
  },
  'Subscript': {
    comp: inlineFormat_1.InlineFormat,
    props: toolsSettings_1.EditorToolsSettings.subscript
  },
  'Superscript': {
    comp: inlineFormat_1.InlineFormat,
    props: toolsSettings_1.EditorToolsSettings.superscript
  },
  'AlignLeft': {
    comp: align_1.Align,
    props: toolsSettings_1.EditorToolsSettings.alignLeft
  },
  'AlignCenter': {
    comp: align_1.Align,
    props: toolsSettings_1.EditorToolsSettings.alignCenter
  },
  'AlignRight': {
    comp: align_1.Align,
    props: toolsSettings_1.EditorToolsSettings.alignRight
  },
  'AlignJustify': {
    comp: align_1.Align,
    props: toolsSettings_1.EditorToolsSettings.alignJustify
  },
  'Indent': {
    comp: indent_1.Indent,
    props: toolsSettings_1.EditorToolsSettings.indent
  },
  'Outdent': {
    comp: outdent_1.Outdent,
    props: toolsSettings_1.EditorToolsSettings.outdent
  },
  'OrderedList': {
    comp: lists_1.List,
    props: toolsSettings_1.EditorToolsSettings.orderedList
  },
  'UnorderedList': {
    comp: lists_1.List,
    props: toolsSettings_1.EditorToolsSettings.bulletList
  },
  'FontSize': {
    comp: fontStyle_1.FontName,
    props: toolsSettings_1.EditorToolsSettings.fontSize
  },
  'FontName': {
    comp: fontStyle_1.FontName,
    props: toolsSettings_1.EditorToolsSettings.fontName
  },
  'FormatBlock': {
    comp: formatBlock_1.FormatBlock,
    props: toolsSettings_1.EditorToolsSettings.formatBlock
  },
  'Undo': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.undo
  },
  'Redo': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.redo
  },
  'Link': {
    comp: insertLink_2.LinkTool,
    props: toolsSettings_1.EditorToolsSettings.link
  },
  'Unlink': {
    comp: unlink_1.Unlink,
    props: toolsSettings_1.EditorToolsSettings.unlink
  },
  'InsertImage': {
    comp: insertImage_1.InsertImage,
    props: toolsSettings_1.EditorToolsSettings.image
  },
  'ViewHtml': {
    comp: viewHtml_1.ViewHtml,
    props: toolsSettings_1.EditorToolsSettings.viewHtml
  },
  'CleanFormatting': {
    comp: cleanFormatting_1.CleanFormatting,
    props: toolsSettings_1.EditorToolsSettings.cleanFormatting
  },
  'SelectAll': {
    comp: selectAll_1.SelectAll,
    props: toolsSettings_1.EditorToolsSettings.selectAll
  },
  'InsertTable': {
    comp: tool_1.InsertTable,
    props: toolsSettings_1.EditorToolsSettings.insertTable
  },
  'MergeCells': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.mergeCells
  },
  'SplitCell': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.splitCell
  },
  'AddRowBefore': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.addRowBefore
  },
  'AddRowAfter': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.addRowAfter
  },
  'AddColumnBefore': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.addColumnBefore
  },
  'AddColumnAfter': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.addColumnAfter
  },
  'DeleteRow': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.deleteRow
  },
  'DeleteColumn': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.deleteColumn
  },
  'DeleteTable': {
    comp: proseMirrorTool_1.ProseMirror,
    props: toolsSettings_1.EditorToolsSettings.deleteTable
  },
  'Print': {
    comp: print_1.Print,
    props: toolsSettings_1.EditorToolsSettings.print
  },
  'Pdf': {
    comp: pdf_1.Pdf,
    props: toolsSettings_1.EditorToolsSettings.pdf
  },
  'InsertFile': {
    comp: insertLink_2.LinkTool,
    props: toolsSettings_1.EditorToolsSettings.insertFile
  },
  'FindAndReplace': {
    comp: findReplace_1.FindAndReplace,
    props: toolsSettings_1.EditorToolsSettings.findAndReplace
  },
  'ForeColor': {
    comp: applyColor_1.ApplyColor,
    props: toolsSettings_1.EditorToolsSettings.foreColor
  },
  'BackColor': {
    comp: applyColor_1.ApplyColor,
    props: toolsSettings_1.EditorToolsSettings.backColor
  }
};
/**
 * @hidden
 */
var EditorVue2 = {
  name: 'KendoEditor',
  // @ts-ignore
  emits: {
    focus: null,
    blur: null,
    change: null,
    loaded: null,
    execute: null
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  props: {
    defaultContent: String,
    value: [Object, String],
    defaultEditMode: {
      type: String,
      default: 'iframe',
      validator: function validator(value) {
        return ['iframe', 'div'].includes(value);
      }
    },
    contentStyle: Object,
    dir: String,
    tools: Array,
    keyboardNavigation: {
      type: Boolean,
      default: true
    },
    resizable: Boolean,
    preserveWhitespace: {
      type: [String, Boolean],
      default: 'full',
      validator: function validator(value) {
        return [true, false, 'full'].includes(value);
      }
    },
    pasteHtml: Function,
    extendView: Function,
    ariaDescribedBy: String,
    ariaLabelledBy: String,
    ariaLabel: String
  },
  data: function data() {
    return {
      updateGuid: (0, kendo_vue_common_1.guid)(),
      view: undefined,
      linkDialog: false
    };
  },
  created: function created() {
    this._view = undefined;
    this.trOnChange = null;
    this.valueisUpdated = false;
    this._prevValue = this.$props.value;
    (0, kendo_vue_common_1.validatePackage)(package_metadata_1.packageMetadata);
  },
  mounted: function mounted() {
    this.iframe = (0, kendo_vue_common_1.getRef)(this, 'iframe');
    this.contentElement = (0, kendo_vue_common_1.getRef)(this, 'contentElement');
    if (!this.iframe || !browser_detection_1.firefox) {
      this.initialize();
    }
  },
  watch: {
    value: function value(_newNextMatch, oldMatch) {
      this.valueisUpdated = true;
      this._prevValue = oldMatch;
    }
  },
  updated: function updated() {
    var value = this.$props.value;
    var view = this.getView();
    if (value === undefined || !this.valueisUpdated || !view) {
      return;
    }
    (0, controlled_value_1.updateEditorValue)(view, this.computedValue(), this._prevValue, this.trOnChange, this.htmlOnChange);
    this.valueisUpdated = false;
    this.trOnChange = null;
    this.htmlOnChange = null;
  },
  destroyed: !!isV3 ? undefined : function () {
    if (this.getView()) {
      this.getView().destroy();
    }
    this.view = undefined;
    this._view = undefined;
  },
  // @ts-ignore
  unmounted: function unmounted() {
    if (this.getView()) {
      this.getView().destroy();
    }
    this.view = undefined;
    this._view = undefined;
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var toolindex = 100;
    var _a = this.$props,
      _b = _a.tools,
      tools = _b === void 0 ? [] : _b,
      _c = _a.defaultEditMode,
      defaultEditMode = _c === void 0 ? 'iframe' : _c,
      _d = _a.preserveWhitespace,
      preserveWhitespace = _d === void 0 ? 'full' : _d,
      style = _a.style,
      value = _a.value;
    var view = this.getView();
    var ls = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    if (this.view) {
      var editorProps = props_key_1.editorPropsKey.getState(this.view.state);
      editorProps.preserveWhitespace = preserveWhitespace;
    }
    var cs = this.$props.contentStyle;
    // vue 2 needs the value inside render in order to update correctly
    var contentStyle = this.v3 ? cs : value ? cs : cs;
    if (contentStyle === undefined && (style || {}).height === undefined) {
      contentStyle = {
        height: '300px'
      };
    }
    var renderDialog = function renderDialog() {
      return this.linkDialog &&
      // @ts-ignore function children
      h(insertLink_1.InsertLinkDialog, {
        view: view,
        attrs: this.v3 ? undefined : {
          view: view,
          settings: link,
          dir: this.$props.dir
        },
        settings: link,
        dir: this.$props.dir,
        onClose: this.handleClose,
        on: this.v3 ? undefined : {
          "close": this.handleClose
        }
      });
    };
    var renderTool = function renderTool(currentTool, _index) {
      var ct = exports.allTools[currentTool] || currentTool;
      var getAriaLabel = function getAriaLabel() {
        var ariaLabel;
        if (currentTool === 'ForeColor') {
          ariaLabel = ls.toLanguageString(main_2.keys.foregroundColorAriaLabel, main_2.messages[main_2.keys.foregroundColorAriaLabel]);
        } else if (currentTool === 'BackColor') {
          ariaLabel = ls.toLanguageString(main_2.keys.backgroundColor, main_2.messages[main_2.keys.backgroundColor]);
        }
        return ariaLabel;
      };
      if (exports.allTools[currentTool]) {
        toolindex++;
        var toolProps = __assign(__assign({
          view: view,
          dir: this.$props.dir,
          key: toolindex,
          updateGuid: this.updateGuid,
          settings: exports.allTools[currentTool].props
        }, exports.allTools[currentTool].props), {
          ariaLabel: getAriaLabel()
        });
        return h(this.v3 ? markRaw(exports.allTools[currentTool].comp) : exports.allTools[currentTool].comp, this.v3 ? toolProps : {
          props: toolProps
        });
      } else if (ct === 'Separator') {
        return (
          // @ts-ignore
          h(kendo_vue_buttons_1.ToolbarSeparator, {
            key: toolindex
          })
        );
      } else {
        var tool = kendo_vue_common_1.templateRendering.call(this, ct.render, kendo_vue_common_1.getListeners.call(this));
        return kendo_vue_common_1.getTemplate.call(this, {
          h: h,
          template: tool,
          defaultRendering: null,
          additionalListeners: {},
          additionalProps: {
            view: view,
            dir: this.$props.dir,
            updateGuid: this.updateGuid,
            key: toolindex,
            settings: ct.props
          }
        });
      }
    };
    var renderTools = function renderTools(item, index) {
      return item.map(function (it) {
        return renderTool.call(this, it, index);
      }, this);
    };
    var buttons = tools.map(function (item, index) {
      var _this2 = this;
      return Array.isArray(item) ?
      // @ts-ignore function children
      h(kendo_vue_buttons_1.ButtonGroup, {
        key: index
      }, this.v3 ? function () {
        return [renderTools.call(_this2, item, index)];
      } : [renderTools.call(_this2, item, index)]) : renderTool.call(this, item, index);
    }, this);
    return h("div", {
      "class": (0, kendo_vue_common_1.classNames)('k-editor', {
        'k-editor-resizable': this.$props.resizable
      }),
      dir: this.$props.dir,
      attrs: this.v3 ? undefined : {
        dir: this.$props.dir
      }
    }, [buttons.length > 0 &&
    // @ts-ignore function children
    h(kendo_vue_buttons_1.Toolbar, {
      "class": 'k-editor-toolbar',
      keyboardNavigation: this.$props.keyboardNavigation,
      attrs: this.v3 ? undefined : {
        keyboardNavigation: this.$props.keyboardNavigation
      }
    }, this.v3 ? function () {
      return [buttons];
    } : [buttons]), defaultEditMode === 'iframe' ? h("div", {
      "class": "k-editor-content"
    }, [h("iframe", {
      onLoad: this.iframeLoad,
      on: this.v3 ? undefined : {
        "load": this.iframeLoad
      },
      ref: (0, kendo_vue_common_1.setRef)(this, 'iframe'),
      frameborder: "0",
      attrs: this.v3 ? undefined : {
        frameborder: "0",
        title: ls.toLanguageString(main_2.keys.iframeTitle, main_2.messages[main_2.keys.iframeTitle])
      },
      title: ls.toLanguageString(main_2.keys.iframeTitle, main_2.messages[main_2.keys.iframeTitle]),
      style: contentStyle,
      "class": "k-iframe"
    })]) : h("div", {
      style: contentStyle,
      "class": "k-editor-content"
    }, [h("div", {
      ref: (0, kendo_vue_common_1.setRef)(this, 'contentElement'),
      role: "textbox",
      attrs: this.v3 ? undefined : {
        role: "textbox",
        "aria-labelledby": this.$props.ariaLabelledBy,
        "aria-describedby": this.$props.ariaDescribedBy,
        "aria-label": this.$props.ariaLabel
      },
      "aria-labelledby": this.$props.ariaLabelledBy,
      "aria-describedby": this.$props.ariaDescribedBy,
      "aria-label": this.$props.ariaLabel
    })]), renderDialog.call(this)]);
  },
  methods: {
    getView: function getView() {
      return this.view;
    },
    getHTML: function getHTML() {
      var view = this.getView();
      if (view) {
        return main_1.EditorUtils.getHtml(view.state);
      }
      return '';
    },
    setHTML: function setHTML(value) {
      var view = this.getView();
      if (view) {
        main_1.EditorUtils.setHtml(view, value);
      }
    },
    focus: function focus() {
      if (this.getView()) {
        this.getView().focus();
      }
    },
    updateTools: function updateTools(view) {
      if (this.v3) {
        this.view = view;
      } else {
        this._view = view;
      }
      if (this.v3) {
        this.updateGuid = (0, kendo_vue_common_1.guid)();
      }
    },
    iframeLoad: function iframeLoad() {
      if (browser_detection_1.firefox) {
        this.initialize();
      }
    },
    initialize: function initialize() {
      var _this = this;
      var iframeWindow = this.iframe && this.iframe.contentWindow;
      if (iframeWindow) {
        var iframeDocument_1 = iframeWindow.document;
        [defaultStyles_1.defaultStyle, defaultStyles_1.tablesStyles, this.$props.dir === 'rtl' ? defaultStyles_1.rtlStyles : undefined].forEach(function (styles) {
          if (styles) {
            var style = iframeDocument_1.createElement('style');
            style.appendChild(iframeDocument_1.createTextNode(styles));
            iframeDocument_1.head.appendChild(style);
          }
        });
        var meta = iframeDocument_1.createElement('meta');
        meta.setAttribute('charset', 'utf-8');
        iframeDocument_1.head.appendChild(meta);
        // The content has to be wrapped because the `dropCursor`
        // plugin raises a `body.offsetParent is null` error.
        this.contentElement = iframeDocument_1.createElement('div');
        iframeDocument_1.body.appendChild(this.contentElement);
        this.contentElement.classList.add('k-content');
        this.contentElement.setAttribute('role', 'main');
      }
      var dom = this.contentElement;
      if (!dom) {
        return;
      }
      var _a = this.$props.preserveWhitespace,
        preserveWhitespace = _a === void 0 ? 'full' : _a;
      var target = this;
      var plugins = [
      // https://prosemirror.net/docs/ref/#state.PluginSpec
      new kendo_editor_common_1.Plugin({
        view: function view() {
          return {
            update: _this.updateTools
          };
        },
        key: new kendo_editor_common_1.PluginKey('toolbar-tools-update-plugin')
      }), new kendo_editor_common_1.Plugin({
        filterTransaction: this.filterTransaction,
        key: new kendo_editor_common_1.PluginKey('onExecute-event-plugin')
      }), new kendo_editor_common_1.Plugin({
        key: props_key_1.editorPropsKey,
        state: {
          init: function init() {
            return {
              preserveWhitespace: preserveWhitespace
            };
          },
          apply: function apply(_, val) {
            return val;
          }
        }
      }), (0, kendo_editor_common_1.spacesFix)(), (0, kendo_editor_common_1.history)(), (0, kendo_editor_common_1.dropCursor)(), (0, kendo_editor_common_1.gapCursor)(), (0, kendo_editor_common_1.tableEditing)()];
      var shortcuts = __assign(__assign({}, main_1.EditorUtils.getShortcuts({
        types: {
          listItem: 'list_item',
          hardBreak: 'hard_break'
        },
        toolsSettings: {
          bold: bold,
          italic: italic,
          underline: underline
        }
      })), {
        'Mod-k': function ModK() {
          var linkDialog = _this.$data.linkDialog;
          var editorView = _this.getView();
          if (editorView) {
            var editorState = editorView.state;
            var collapsed = editorState.selection.empty;
            var linkMark = (0, kendo_editor_common_1.getMark)(editorState, editorState.schema.marks[link.mark]);
            var disabled = collapsed && !linkMark;
            if (!linkDialog && !disabled) {
              _this.linkDialog = true;
            }
          }
          return !linkDialog;
        }
      });
      var _b = this.$props,
        _c = _b.defaultContent,
        defaultContent = _c === void 0 ? '' : _c,
        value = _b.value;
      var doc = value && typeof value !== 'string' ? this.v3 ? toRaw(value) : value : main_1.EditorUtils.createDocument(new kendo_editor_common_1.Schema({
        nodes: schema_1.nodes,
        marks: schema_1.marks
      }), value || defaultContent, {
        preserveWhitespace: preserveWhitespace
      });
      var viewProps = {
        state: kendo_editor_common_1.EditorState.create({
          plugins: __spreadArray(__spreadArray([], plugins, true), [(0, kendo_editor_common_1.keymap)(shortcuts), (0, kendo_editor_common_1.keymap)(kendo_editor_common_1.baseKeymap)], false),
          doc: doc
        }),
        transformPastedHTML: this.onPasteHtml,
        dispatchTransaction: this.dispatchTransaction,
        handleDOMEvents: {
          focus: this.onFocus,
          blur: this.onBlur,
          paste: this.onPaste
        }
      };
      var mountEvent = {
        plugins: plugins,
        shortcuts: shortcuts,
        target: target,
        viewProps: viewProps,
        dom: dom
      };
      this.$emit('loaded', mountEvent);
      var view = this.$props.extendView && this.$props.extendView(mountEvent) || new kendo_editor_common_1.EditorView({
        mount: dom
      }, viewProps);
      this.view = this.v3 ? markRaw(view) : view;
    },
    filterTransaction: function filterTransaction(transaction, state) {
      var event = {
        target: this,
        transaction: transaction,
        state: state
      };
      this.$emit('execute', event);
      return true;
    },
    onPasteHtml: function onPasteHtml(html) {
      if (this.$props.pasteHtml && this.pasteEvent) {
        var event_1 = {
          target: this,
          pastedHtml: html,
          event: this.pasteEvent
        };
        var newHtml = this.$props.pasteHtml(event_1);
        this.pasteEvent = undefined;
        if (typeof newHtml === 'string') {
          return newHtml;
        }
      }
      return html;
    },
    dispatchTransaction: function dispatchTransaction(transaction) {
      var docChanged = transaction.docChanged;
      if (docChanged) {
        this.trOnChange = transaction;
        var doc_1 = transaction.doc,
          schema_2 = transaction.doc.type.schema;
        var target_1 = this;
        var event_2 = {
          target: target_1,
          value: doc_1,
          get html() {
            target_1.htmlOnChange = main_1.EditorUtils.getHtml({
              doc: doc_1,
              schema: schema_2
            });
            return target_1.htmlOnChange;
          },
          transaction: transaction,
          schema: schema_2
        };
        this.$emit('change', event_2);
      }
      if (this.getView() && (this.$props.value === undefined || !docChanged)) {
        this.getView().updateState(this.getView().state.apply(transaction));
      }
    },
    onFocus: function onFocus(_view, e) {
      var event = {
        target: this,
        event: e
      };
      this.$emit('focus', event);
      return false;
    },
    onBlur: function onBlur(_view, e) {
      var event = {
        target: this,
        event: e
      };
      this.$emit('blur', event);
      return false;
    },
    onPaste: function onPaste(_view, nativeEvent) {
      if (this.$props.pasteHtml) {
        this.pasteEvent = nativeEvent;
      }
      return false;
    },
    handleClose: function handleClose() {
      this.linkDialog = false;
    },
    computedValue: function computedValue() {
      if (this.trOnChange !== null) {
        return this.trOnChange.doc;
      } else if (this.$props.value !== undefined) {
        return this.$props.value;
      } else if (this.getView()) {
        return this.getView().state.doc;
      }
      return this.$props.defaultContent || '';
    }
  }
};
exports.EditorVue2 = EditorVue2;
/**
 * Represents the [Kendo UI for Vue Editor component]({% slug overview_editor %}).
 *
 *  ```jsx
 *  <template>
 *    <div>
 *      <Editor
 *        :tools="tools"
 *        :content-style="{
 *          height: '690px',
 *        }"
 *        :default-content="content"
 *      />
 *    </div>
 *  </template>
 *
 *  <script>
 *  import { Editor } from "@progress/kendo-vue-editor";
 *  import content from "./content-overview";
 *
 *  export default {
 *    components: {
 *      Editor,
 *    },
 *    data() {
 *      return {
 *        tools: [
 *          ["Bold", "Italic", "Underline", "Strikethrough"],
 *          ["Subscript", "Superscript"],
 *          ["AlignLeft", "AlignCenter", "AlignRight", "AlignJustify"],
 *          ["Indent", "Outdent"],
 *          ["OrderedList", "UnorderedList"],
 *          "FontSize",
 *          "FontName",
 *          "FormatBlock",
 *          ["Undo", "Redo"],
 *          ["Link", "Unlink", "InsertImage", "ViewHtml"],
 *          ["InsertTable"],
 *          ["AddRowBefore", "AddRowAfter", "AddColumnBefore", "AddColumnAfter"],
 *          ["DeleteRow", "DeleteColumn", "DeleteTable"],
 *          ["MergeCells", "SplitCell"],
 *        ],
 *        content: content,
 *      };
 *    },
 *  };
 *  </script>
 *  ```
 *
 *
 *
 *  ### props <span class='code'>Readonly&lt;[EditorProps]({% slug api_editor_editorprops %})</span>
 *  The props of the Editor component.
 *
 *  ### contentElement <span class='code'>HTMLDivElement</span>
 *  Returns the content-editable DOM element of the Editor.
 *
 *  ### element <span class='code'>HTMLElement</span>
 *  Returns the DOM element of the Editor.
 *
 *  ### value <span class='code'>Node | string</span>
 *  The value of the Editor.
 *
 *  ### view <span class='code'>EditorView&lt;any&gt;</span>
 *  Returns the `view` object of the Editor.
 */
var Editor = EditorVue2;
exports.Editor = Editor;
