// @progress/kendo-vue-scheduler/dist/npm/views/time/TimelineViewAllEventsRowContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TimelineViewAllEventsRowContentVue2 = exports.TimelineViewAllEventsRowContent = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var main_1 = require("../../messages/main");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
/** @hidden */
var TimelineViewAllEventsRowContentVue2 = {
  name: 'KendoTimelineViewAllEventsRowContent',
  props: {},
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    return h("div", {
      "class": "k-sticky-cell"
    }, [h("div", {
      "class": "k-scheduler-cell k-heading-cell k-side-cell k-scheduler-times-all-day",
      "data-range-index": 0,
      attrs: this.v3 ? undefined : {
        "data-range-index": 0
      }
    }, [localization.toLanguageString(main_1.allEvents, main_1.messages[main_1.allEvents])])]);
  }
};
exports.TimelineViewAllEventsRowContentVue2 = TimelineViewAllEventsRowContentVue2;
/** @hidden */
var TimelineViewAllEventsRowContent = TimelineViewAllEventsRowContentVue2;
exports.TimelineViewAllEventsRowContent = TimelineViewAllEventsRowContent;
