// @progress/kendo-scheduler-vue-wrapper/dist/es/KendoScheduler/KendoSchedulerMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _kendoBaseComponentsVueWrapper = require('@progress/kendo-base-components-vue-wrapper');

var _SchedulerProps = require('./SchedulerProps');

var _SchedulerProps2 = _interopRequireDefault(_SchedulerProps);

var _KendoSchedulerResource = require('../KendoSchedulerResource');

var _KendoSchedulerResource2 = _interopRequireDefault(_KendoSchedulerResource);

var _KendoSchedulerView = require('../KendoSchedulerView');

var _KendoSchedulerView2 = _interopRequireDefault(_KendoSchedulerView);

var _CurrentTimeMarker = require('./CurrentTimeMarker');

var _CurrentTimeMarker2 = _interopRequireDefault(_CurrentTimeMarker);

var _Editable = require('./Editable');

var _Editable2 = _interopRequireDefault(_Editable);

var _Footer = require('./Footer');

var _Footer2 = _interopRequireDefault(_Footer);

var _Group = require('./Group');

var _Group2 = _interopRequireDefault(_Group);

var _Messages = require('./Messages');

var _Messages2 = _interopRequireDefault(_Messages);

var _Pdf = require('./Pdf');

var _Pdf2 = _interopRequireDefault(_Pdf);

var _Toolbar = require('./Toolbar');

var _Toolbar2 = _interopRequireDefault(_Toolbar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    props: _SchedulerProps2.default,
    emits: {
        'kendowidgetready': null
    },
    mixins: [_kendoBaseComponentsVueWrapper.KendoBaseComponent, _kendoBaseComponentsVueWrapper.KendoBaseDatasourceComponent, _CurrentTimeMarker2.default, _Editable2.default, _Footer2.default, _Group2.default, _Messages2.default, _Pdf2.default, _Toolbar2.default],
    mounted: function mounted() {
        this._resolveChildren();

        kendo.jQuery(this.$el).kendoScheduler(this.widgetOptions);

        if (this.v3) {
            this.ready();
            this.$emit('kendowidgetready', this.kendoWidget());
        } else {
            this.$emit('kendowidgetready', this.kendoWidget());
        }
    },

    methods: {
        kendoWidget: function kendoWidget() {
            return kendo.jQuery(this.$el).getKendoScheduler();
        },
        _resolveChildren: function _resolveChildren() {
            this.resolveChildren('resources', _KendoSchedulerResource2.default.name);
            this.resolveChildren('views', _KendoSchedulerView2.default.name);
        }
    }
};
