// @progress/kendo-vue-scheduler/dist/npm/views/agenda/AgendaViewHead.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AgendaViewHeadVue2 = exports.AgendaViewHead = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var main_1 = require("../../messages/main");
/**
 * @hidden
 */
var AgendaViewHeadVue2 = {
  name: 'KendoAgendaViewHead',
  props: {
    monthSlots: Array,
    dateHeaderCell: [String, Function, Object]
  },
  inject: {
    ri: {
      default: null
    },
    kendoLocalizationService: {
      default: null
    },
    ks: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    return h("div", {
      "class": "k-scheduler-row",
      key: this.ri.groupIndex
    }, [h("div", {
      "class": "k-scheduler-cell k-heading-cell k-group-cell k-scheduler-datecolumn"
    }, [localization.toLanguageString(main_1.dateTitle, main_1.messages[main_1.dateTitle])]), h("div", {
      "class": "k-scheduler-cell k-heading-cell k-group-cell k-scheduler-timecolumn"
    }, [localization.toLanguageString(main_1.timeTitle, main_1.messages[main_1.timeTitle])]), h("div", {
      "class": "k-scheduler-cell k-heading-cell k-scheduler-eventcolumn"
    }, [localization.toLanguageString(main_1.eventTitle, main_1.messages[main_1.eventTitle])])]);
  }
};
exports.AgendaViewHeadVue2 = AgendaViewHeadVue2;
/**
 * @hidden
 */
var AgendaViewHead = AgendaViewHeadVue2;
exports.AgendaViewHead = AgendaViewHead;
