// @progress/kendo-vue-scheduler/dist/npm/views/agenda/AgendaViewHeaderItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AgendaViewHeaderItemVue2 = exports.AgendaViewHeaderItem = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var inject = allVue.inject;
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
/**
 * @hidden
 */
var AgendaViewHeaderItemVue2 = {
  name: 'KendoAgendaViewHeaderItem',
  props: {
    range: Object
  },
  inject: {
    kendoIntlService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var intl = (0, kendo_vue_intl_1.provideIntlService)(this);
    return h("div", {
      "class": "k-scheduler-cell k-scheduler-datecolumn k-group-cell"
    }, [h("div", [h("strong", {
      "class": "k-scheduler-agendaday"
    }, [intl.formatDate(this.$props.range.zonedStart, 'dd')]), h("em", {
      "class": "k-scheduler-agendaweek"
    }, [intl.formatDate(this.$props.range.zonedStart, 'EEEE')]), h("span", {
      "class": "k-scheduler-agendadate"
    }, [intl.formatDate(this.$props.range.zonedStart, 'y')])])]);
  }
};
exports.AgendaViewHeaderItemVue2 = AgendaViewHeaderItemVue2;
/**
 * @hidden
 */
var AgendaViewHeaderItem = AgendaViewHeaderItemVue2;
exports.AgendaViewHeaderItem = AgendaViewHeaderItem;
