// @progress/kendo-vue-scheduler/dist/npm/components/header/navigation/SchedulerNavigation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SchedulerNavigationVue2 = exports.SchedulerNavigation = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");
var kendo_vue_buttons_2 = require("@progress/kendo-vue-buttons");
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
var main_1 = require("../../../messages/main");
var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");
var kendo_svg_icons_1 = require("@progress/kendo-svg-icons");
/**
 * @hidden
 */
var SchedulerNavigationVue2 = {
  name: 'KendoSchedulerNavigation',
  props: {
    isRtl: Boolean
  },
  inject: {
    ks: {
      default: null
    },
    kendoLocalizationService: {
      default: null
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var _this = this;
    var h = gh || createElement;
    var defaultSlot = (0, kendo_vue_common_1.getDefaultSlots)(this);
    this.localization = (0, kendo_vue_intl_1.provideLocalizationService)(this);
    var todayText = this.localization.toLanguageString(main_1.today, main_1.messages[main_1.today]);
    var previousText = this.localization.toLanguageString(main_1.previousTitle, main_1.messages[main_1.previousTitle]);
    var nextText = this.localization.toLanguageString(main_1.nextTitle, main_1.messages[main_1.nextTitle]);
    return (
      // @ts-ignore  function children
      h(kendo_vue_buttons_2.ToolbarItem, this.v3 ? function () {
        return [
        // @ts-ignore  function children
        h(kendo_vue_buttons_1.ButtonGroup, {
          "class": "k-scheduler-navigation"
        }, _this.v3 ? function () {
          return [
          // @ts-ignore  function children
          h(kendo_vue_buttons_1.Button, {
            role: "button",
            attrs: _this.v3 ? undefined : {
              role: "button",
              tabIndex: -1,
              title: todayText,
              "aria-label": todayText
            },
            "class": "k-nav-today",
            tabIndex: -1,
            title: todayText,
            "aria-label": todayText,
            onClick: _this.handleTodayClick,
            on: _this.v3 ? undefined : {
              "click": _this.handleTodayClick
            }
          }, _this.v3 ? function () {
            return [todayText];
          } : [todayText]),
          // @ts-ignore  function children
          h(kendo_vue_buttons_1.Button, {
            role: "button",
            attrs: _this.v3 ? undefined : {
              role: "button",
              tabIndex: -1,
              icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
              svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
              title: previousText,
              "aria-label": previousText
            },
            tabIndex: -1,
            "class": "k-button-icon k-nav-prev",
            icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
            svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
            title: previousText,
            "aria-label": previousText,
            onClick: _this.handlePrevClick,
            on: _this.v3 ? undefined : {
              "click": _this.handlePrevClick
            }
          }),
          // @ts-ignore  function children
          h(kendo_vue_buttons_1.Button, {
            role: "button",
            attrs: _this.v3 ? undefined : {
              role: "button",
              tabIndex: -1,
              icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
              svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
              title: nextText,
              "aria-label": nextText
            },
            tabIndex: -1,
            "class": "k-button-icon k-nav-next",
            icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
            svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
            title: nextText,
            "aria-label": nextText,
            onClick: _this.handleNextClick,
            on: _this.v3 ? undefined : {
              "click": _this.handleNextClick
            }
          })];
        } : [h(kendo_vue_buttons_1.Button, {
          role: "button",
          attrs: _this.v3 ? undefined : {
            role: "button",
            tabIndex: -1,
            title: todayText,
            "aria-label": todayText
          },
          "class": "k-nav-today",
          tabIndex: -1,
          title: todayText,
          "aria-label": todayText,
          onClick: _this.handleTodayClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleTodayClick
          }
        }, _this.v3 ? function () {
          return [todayText];
        } : [todayText]), h(kendo_vue_buttons_1.Button, {
          role: "button",
          attrs: _this.v3 ? undefined : {
            role: "button",
            tabIndex: -1,
            icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
            svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
            title: previousText,
            "aria-label": previousText
          },
          tabIndex: -1,
          "class": "k-button-icon k-nav-prev",
          icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
          svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
          title: previousText,
          "aria-label": previousText,
          onClick: _this.handlePrevClick,
          on: _this.v3 ? undefined : {
            "click": _this.handlePrevClick
          }
        }), h(kendo_vue_buttons_1.Button, {
          role: "button",
          attrs: _this.v3 ? undefined : {
            role: "button",
            tabIndex: -1,
            icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
            svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
            title: nextText,
            "aria-label": nextText
          },
          tabIndex: -1,
          "class": "k-button-icon k-nav-next",
          icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
          svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
          title: nextText,
          "aria-label": nextText,
          onClick: _this.handleNextClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleNextClick
          }
        })])];
      } : [h(kendo_vue_buttons_1.ButtonGroup, {
        "class": "k-scheduler-navigation"
      }, _this.v3 ? function () {
        return [h(kendo_vue_buttons_1.Button, {
          role: "button",
          attrs: _this.v3 ? undefined : {
            role: "button",
            tabIndex: -1,
            title: todayText,
            "aria-label": todayText
          },
          "class": "k-nav-today",
          tabIndex: -1,
          title: todayText,
          "aria-label": todayText,
          onClick: _this.handleTodayClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleTodayClick
          }
        }, _this.v3 ? function () {
          return [todayText];
        } : [todayText]), h(kendo_vue_buttons_1.Button, {
          role: "button",
          attrs: _this.v3 ? undefined : {
            role: "button",
            tabIndex: -1,
            icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
            svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
            title: previousText,
            "aria-label": previousText
          },
          tabIndex: -1,
          "class": "k-button-icon k-nav-prev",
          icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
          svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
          title: previousText,
          "aria-label": previousText,
          onClick: _this.handlePrevClick,
          on: _this.v3 ? undefined : {
            "click": _this.handlePrevClick
          }
        }), h(kendo_vue_buttons_1.Button, {
          role: "button",
          attrs: _this.v3 ? undefined : {
            role: "button",
            tabIndex: -1,
            icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
            svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
            title: nextText,
            "aria-label": nextText
          },
          tabIndex: -1,
          "class": "k-button-icon k-nav-next",
          icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
          svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
          title: nextText,
          "aria-label": nextText,
          onClick: _this.handleNextClick,
          on: _this.v3 ? undefined : {
            "click": _this.handleNextClick
          }
        })];
      } : [h(kendo_vue_buttons_1.Button, {
        role: "button",
        attrs: _this.v3 ? undefined : {
          role: "button",
          tabIndex: -1,
          title: todayText,
          "aria-label": todayText
        },
        "class": "k-nav-today",
        tabIndex: -1,
        title: todayText,
        "aria-label": todayText,
        onClick: _this.handleTodayClick,
        on: _this.v3 ? undefined : {
          "click": _this.handleTodayClick
        }
      }, _this.v3 ? function () {
        return [todayText];
      } : [todayText]), h(kendo_vue_buttons_1.Button, {
        role: "button",
        attrs: _this.v3 ? undefined : {
          role: "button",
          tabIndex: -1,
          icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
          svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
          title: previousText,
          "aria-label": previousText
        },
        tabIndex: -1,
        "class": "k-button-icon k-nav-prev",
        icon: _this.isRtl ? 'caret-alt-right' : 'caret-alt-left',
        svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltRightIcon : kendo_svg_icons_1.caretAltLeftIcon,
        title: previousText,
        "aria-label": previousText,
        onClick: _this.handlePrevClick,
        on: _this.v3 ? undefined : {
          "click": _this.handlePrevClick
        }
      }), h(kendo_vue_buttons_1.Button, {
        role: "button",
        attrs: _this.v3 ? undefined : {
          role: "button",
          tabIndex: -1,
          icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
          svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
          title: nextText,
          "aria-label": nextText
        },
        tabIndex: -1,
        "class": "k-button-icon k-nav-next",
        icon: _this.isRtl ? 'caret-alt-left' : 'caret-alt-right',
        svgIcon: _this.isRtl ? kendo_svg_icons_1.caretAltLeftIcon : kendo_svg_icons_1.caretAltRightIcon,
        title: nextText,
        "aria-label": nextText,
        onClick: _this.handleNextClick,
        on: _this.v3 ? undefined : {
          "click": _this.handleNextClick
        }
      })])])
    );
  },
  methods: {
    handleTodayClick: function handleTodayClick(e) {
      this.$emit('todayclick', e);
    },
    handlePrevClick: function handlePrevClick(e) {
      this.$emit('prevclick', e);
    },
    handleNextClick: function handleNextClick(e) {
      this.$emit('nextclick', e);
    }
  }
};
exports.SchedulerNavigationVue2 = SchedulerNavigationVue2;
/**
 * @hidden
 */
var SchedulerNavigation = SchedulerNavigationVue2;
exports.SchedulerNavigation = SchedulerNavigation;
