// @progress/kendo-vue-scheduler/dist/npm/context/SchedulerResourceIteratorContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SchedulerResourceIteratorContextVue2 = exports.SchedulerResourceIteratorContext = void 0;
// @ts-ignore
var Vue = require("vue");
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
var kendo_vue_common_1 = require("@progress/kendo-vue-common");
/** @hidden */
var SchedulerResourceIteratorContextVue2 = {
  name: 'KendoSchedulerResourceIteratorContext',
  inheritAttrs: false,
  props: {
    noWrap: Boolean,
    resource: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    groupIndex: {
      type: Number,
      default: 0
    }
  },
  data: function data() {
    return {
      ri: {
        resource: this.resource,
        groupIndex: this.groupIndex
      }
    };
  },
  provide: function provide() {
    return {
      ri: this.ri
    };
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlot = (0, kendo_vue_common_1.getDefaultSlots)(this);
    return defaultSlot ? this.$props.noWrap && defaultSlot ? defaultSlot[0] : h("div", {
      "class": "k-scheduler-cell k-group-cell"
    }, [defaultSlot]) : null;
  }
};
exports.SchedulerResourceIteratorContextVue2 = SchedulerResourceIteratorContextVue2;
/** @hidden */
var SchedulerResourceIteratorContext = SchedulerResourceIteratorContextVue2;
exports.SchedulerResourceIteratorContext = SchedulerResourceIteratorContext;
